# Bioinformatics resources

## Gene Ontology

One important difficulty when dealing with Biological structures (molecules, organelles, etc.) is the vocabulary we use to described them. As this type of information accumulated different researches would tend to use slighly different ways to describe the same thing. In other cases they can use similar words to describe different things. These nuances make the descriptions often found in publications ambiguous making it difficult to determine the exact meaning of some assertion. This difficulty became more prominent once the amount of information increased dramatically with the advent of omics technologies, and the information about the biological stuff was the subject of computational analyses. There was a need for an standardized nomenclature, and that was the origin of the [Gene Ontology](http://www.geneontology.org) project. The purpose of the Gene Ontology project is stated in it web page:

>The Gene Ontology (GO) project is a major bioinformatics initiative to develop a computational representation of our evolving knowledge of how genes encode biological functions at the molecular, cellular and tissue system levels. Biological systems are so complex that we need to rely on computers to represent this knowledge. The project has developed formal ontologies that represent over 40,000 biological concepts, and are constantly being revised to reflect new discoveries. To date, these concepts have been used to "annotate" gene functions based on experiments reported in over 100,000 peer-reviewed scientific papers.

In particular:

>The Gene Ontology project provides a controlled vocabulary of terms for describing gene product characteristics and gene product annotation data from GO Consortium members, as well as tools to access and process these data.

```{r go-web, echo=FALSE, fig.cap="Screenshot of the Gene Ontology home page at http://www.geneontology.org."}
knitr::include_graphics("pic/go_web.png")
```

## Sequence databases

### NCBI

### EnsEMBL

### Uniprot

## Pathway databases

As information about genes, proteins and metabolites accumulates, it becomes more and more important to organize this knowledge in a comprehensive way. We have mentioned how genes and their products participate in the networks of cellular processes, the cell's pathway. This network includes metabolic, signaling and gene regulatory networks. This information is critical to understand the cell's phenotype, but the accumulated knowledge is daunting. It would benefit to have some central repository with easily accessible tools that enables to visualize existing knowledge in an easy way. It would be also very useful if it was possible to use that information in computer analysis. This is the motivation behind some of the web resources described below.

### KEGG

KEGG stands for [Kyoto Encyclopedia for Genes and Genomes](http://www.kegg.jp) (Figure \@ref(fig:kegg-web)). Is an integrative resource with information about matabolic and signaling pathways in different species, from bacteria and archaea to mammals.

```{r kegg-web, echo=FALSE, fig.cap="Screenshot of the KEGG home page at http://www.kegg.jp."}
knitr::include_graphics("pic/kegg_web.png")
```


### Reactome

## Domain databases

### Pfam

[Pfam](http://pfam.xfam.org) is a database of protein domain families.

>Proteins are generally comprised of one or more functional regions, commonly termed domains. The presence of different domains in varying combinations in different proteins gives rise to the diverse repertoire of proteins found in nature. Identifying the domains present in a protein can provide insights into the function of that protein.
>
>The Pfam database is a large collection of protein domain families. Each family is represented by multiple sequence alignments and a hidden Markov model (HMMs).

```{r pfam-web, echo=FALSE, fig.cap="Screenshot of the Pfam home page at http://pfam.xfam.org."}
knitr::include_graphics("pic/pfam_web.png")
```

## Motif databases

## Interaction networks

Information about protein-protein interactions has grown incredibly due to advances in MS in combination with affinity purification methods. There are several databases integrating existing information. Two resources will be highlighted here: BioGRID and STRING.

### BioGRID

[BioGRID](https://thebiogrid.org) focuses on storing information about PPIs for which experimental evidence exists.

```{r biogrid-web, echo=FALSE, fig.cap="Screenshot of the BioGRID home page at https://thebiogrid.org."}
knitr::include_graphics("pic/biogrid_web.png")
```

### STRING

STRING combines experimental and computational evidence to infer PPIs.

```{r string-web, echo=FALSE, fig.cap="Screenshot of the STRING home page at https://string-db.org."}
knitr::include_graphics("pic/string_web.png")
```

## Bioformatics software

### Blast

The [Blast](https://blast.ncbi.nlm.nih.gov) web page at NCBI is the main point of access to this versitile tool. It can be used to search for sequence similarity against any of the sequence databases at the NCBI.

```{r blast-web, echo=FALSE, fig.cap="Screenshot of the Blast home page at https://blast.ncbi.nlm.nih.gov."}
knitr::include_graphics("pic/blast_web.png")
```

### Bioconductor

### Cytoscape

