# Learning Bioinformatics {#learning-bioinformatics}

## Books

### Sequence analysis

#### BLAST: An essential guide to the Basic Local Alignment Search Tool {-}

Introductory book to the topic of sequence aligments in general and Blast in particular. Includes Perl code.

```{r, echo=FALSE, fig.cap="Blast. Ian Korf, ... O'Reilly, 2003"}
knitr::include_graphics("pic/blast_book.jpg")
```

#### Biological sequence analysis: Probabilistic models of proteins and nucleic acids {-}

Advance level understanding of the theory of sequence aligment and Hidden Markov Models (HMM).

```{r, echo=FALSE, fig.cap="Biological sequence analysis: Probabilistic models of proteins and nucleic acids. R. Durbin, S. Eddy, ... Cambridge, 1998"}
knitr::include_graphics("pic/hmm_book.jpg")
```

#### Molecular evolution and phylogenetics {-}

Introductory level book on the topic of inferring phylogenies from sequence information.

```{r, echo=FALSE, fig.cap="Molecular evolution and phylogenetics. M. Nei, S. Kumar. Oxford University Press, 2000."}
knitr::include_graphics("pic/phylogenetics_book.jpg")
```

### Omics technologies

## Online resources

### Learning how to use the command line

Sometimes using Bioinformatics tools requires to have some knowledge of how to use the command line, specially in Linux/Unix environments. [http://linuxcommand.org](http://linuxcommand.org/index.php) has a nice introduction to the Linux command line.

Another interesting resource is [https://www.codecademy.com](https://www.codecademy.com), providing an interactive learning experience. It offers also paid courses.

### Support Communities

Learning Bioinformatics, as with any other complex topic, can be daunting. Often Bioinformatics software come with good and detailed documentation and. It is also possible to frequently find short tutorials that can help newcomers get started. It is important to get a *good* understanding of the methods being used when the goal is to publish your results. The Bioinformatics community can sometimes help in the learning process by providing solutions to the most frequently encountered problems. Here I will highlight some of the most relevant online communities for the Bioinformatician.

#### Biostars {-}

[Biostars](http://biostars.org)

#### Bioinformatics Stack Exchange {-}

Another useful resource for solving questions related to Bioinformatics might be the [Bioinformatics](https://bioinformatics.stackexchange.com) channel at the Stackexchange network.

#### Bioconductor support site {-}

The [Bioconductor support site](https://support.bioconductor.org) provides support for questions related to the use of Bioconductor in particular and R for Bioinformatics more in general.


#### How to use the support sites {-}

It is important to remember that online support sites are usually driven by voluntarees that lend their time and expertise to help others. Because the number of questions put in one of these support sites can be very large, it is very helpful if the questions themeselves are written in a way that simplify the work of those willing to help us with our doubts. For example, it is important to write simple yet meaningful titles that summarize what our problem is. This way the community can easily know if the question is related to a topic about which they have expertise. When describing your issues, be as explicit as possible. And whenever possible, include a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_Working_Example). This is particularly important if you have problems with programming code but it can also apply to using a web server. For example, if you had some problem using a web server to generate a sequence aligment, include all the steps you performed to obtain the error (if any) you obtained. If possible, when needed, include a minimal example dataset to reproduce the error. In this example, it could be a small file with a few sequences that trigger the error.

The particulars will depend of the software and problem you are having. But as a rule of thumb it is always helpful to put yourself in the place of those reading your question, and imagining whether that is enough for them to understand your problem.
