<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Introduction to Bioinformatics</title>
  <meta name="description" content="Material covered in the <em>Introduction to Bioinformatics</em> intensive course held in Nagoya University, Dec 13-14, 2017.">
  <meta name="generator" content="bookdown 0.5 and GitBook 2.6.7">

  <meta property="og:title" content="Introduction to Bioinformatics" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Material covered in the <em>Introduction to Bioinformatics</em> intensive course held in Nagoya University, Dec 13-14, 2017." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Introduction to Bioinformatics" />
  
  <meta name="twitter:description" content="Material covered in the <em>Introduction to Bioinformatics</em> intensive course held in Nagoya University, Dec 13-14, 2017." />
  

<meta name="author" content="Diego Diez">


<meta name="date" content="2017-12-09">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="sequence-analysis.html">
<link rel="next" href="bioinformatics-resources.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Introduction to Bioinformatics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About</a></li>
<li class="chapter" data-level="" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i>Prerequisites</a></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="intro.html"><a href="intro.html#intro-sequence-analysis"><i class="fa fa-check"></i><b>1.1</b> Sequence analysis</a></li>
<li class="chapter" data-level="1.2" data-path="intro.html"><a href="intro.html#omics-analysis"><i class="fa fa-check"></i><b>1.2</b> Omics analysis</a></li>
<li class="chapter" data-level="1.3" data-path="intro.html"><a href="intro.html#structural-analysis"><i class="fa fa-check"></i><b>1.3</b> Structural analysis</a></li>
<li class="chapter" data-level="1.4" data-path="intro.html"><a href="intro.html#phylogenetic-analysis"><i class="fa fa-check"></i><b>1.4</b> Phylogenetic analysis</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="sequence-analysis.html"><a href="sequence-analysis.html"><i class="fa fa-check"></i><b>2</b> Sequence analysis</a><ul>
<li class="chapter" data-level="2.1" data-path="sequence-analysis.html"><a href="sequence-analysis.html#pairwise-alignment"><i class="fa fa-check"></i><b>2.1</b> Pairwise alignment</a></li>
<li class="chapter" data-level="2.2" data-path="sequence-analysis.html"><a href="sequence-analysis.html#scoring-matrices"><i class="fa fa-check"></i><b>2.2</b> Scoring matrices</a></li>
<li class="chapter" data-level="2.3" data-path="sequence-analysis.html"><a href="sequence-analysis.html#blast"><i class="fa fa-check"></i><b>2.3</b> Blast</a></li>
<li class="chapter" data-level="2.4" data-path="sequence-analysis.html"><a href="sequence-analysis.html#multiple-sequence-alignment"><i class="fa fa-check"></i><b>2.4</b> Multiple sequence alignment</a></li>
<li class="chapter" data-level="2.5" data-path="sequence-analysis.html"><a href="sequence-analysis.html#tree-methods"><i class="fa fa-check"></i><b>2.5</b> Tree methods</a></li>
<li class="chapter" data-level="2.6" data-path="sequence-analysis.html"><a href="sequence-analysis.html#domains-and-motifs"><i class="fa fa-check"></i><b>2.6</b> Domains and motifs</a></li>
<li class="chapter" data-level="2.7" data-path="sequence-analysis.html"><a href="sequence-analysis.html#phylogenetics"><i class="fa fa-check"></i><b>2.7</b> Phylogenetics</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html"><i class="fa fa-check"></i><b>3</b> Omics analysis</a><ul>
<li class="chapter" data-level="3.1" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#microarrays"><i class="fa fa-check"></i><b>3.2</b> Microarrays</a><ul>
<li class="chapter" data-level="3.2.1" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#expression-microarrays"><i class="fa fa-check"></i><b>3.2.1</b> Expression microarrays</a></li>
<li class="chapter" data-level="3.2.2" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#analysis-of-microarray-data"><i class="fa fa-check"></i><b>3.2.2</b> Analysis of microarray data</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#next-generation-sequencing"><i class="fa fa-check"></i><b>3.3</b> Next-generation sequencing</a><ul>
<li class="chapter" data-level="3.3.1" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#rna-seq"><i class="fa fa-check"></i><b>3.3.1</b> RNA-seq</a></li>
<li class="chapter" data-level="3.3.2" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#example-stat3-expression-in-macrophages-exposed-to-il10"><i class="fa fa-check"></i><b>3.3.2</b> Example: Stat3 expression in macrophages exposed to IL10</a></li>
<li class="chapter" data-level="3.3.3" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#chip-seq"><i class="fa fa-check"></i><b>3.3.3</b> ChIP-seq</a></li>
<li class="chapter" data-level="3.3.4" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#example-stat3-binding-in-macrophages-exposed-to-il10"><i class="fa fa-check"></i><b>3.3.4</b> Example: Stat3 binding in macrophages exposed to IL10</a></li>
<li class="chapter" data-level="3.3.5" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#other-ngs-technologies"><i class="fa fa-check"></i><b>3.3.5</b> Other NGS technologies</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#mass-spectromics-methods"><i class="fa fa-check"></i><b>3.4</b> Mass spectromics methods</a><ul>
<li class="chapter" data-level="3.4.1" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#ms"><i class="fa fa-check"></i><b>3.4.1</b> MS</a></li>
<li class="chapter" data-level="3.4.2" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#ap-ms"><i class="fa fa-check"></i><b>3.4.2</b> AP-MS</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#single-cell-omics"><i class="fa fa-check"></i><b>3.5</b> Single cell omics</a></li>
<li class="chapter" data-level="3.6" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#batch-effects"><i class="fa fa-check"></i><b>3.6</b> Batch effects</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html"><i class="fa fa-check"></i><b>4</b> Bioinformatics resources</a><ul>
<li class="chapter" data-level="4.1" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#gene-ontology"><i class="fa fa-check"></i><b>4.1</b> Gene Ontology</a></li>
<li class="chapter" data-level="4.2" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#sequence-databases"><i class="fa fa-check"></i><b>4.2</b> Sequence databases</a><ul>
<li class="chapter" data-level="4.2.1" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#ncbi"><i class="fa fa-check"></i><b>4.2.1</b> NCBI</a></li>
<li class="chapter" data-level="4.2.2" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#ensembl"><i class="fa fa-check"></i><b>4.2.2</b> EnsEMBL</a></li>
<li class="chapter" data-level="4.2.3" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#uniprot"><i class="fa fa-check"></i><b>4.2.3</b> Uniprot</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#gene-expression-databases"><i class="fa fa-check"></i><b>4.3</b> Gene expression databases</a><ul>
<li class="chapter" data-level="4.3.1" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#geo-gene-expression-omnibus"><i class="fa fa-check"></i><b>4.3.1</b> GEO: Gene Expression Omnibus</a></li>
<li class="chapter" data-level="4.3.2" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#arrayexpress"><i class="fa fa-check"></i><b>4.3.2</b> ArrayExpress</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#pathway-databases"><i class="fa fa-check"></i><b>4.4</b> Pathway databases</a><ul>
<li class="chapter" data-level="4.4.1" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#kegg"><i class="fa fa-check"></i><b>4.4.1</b> KEGG</a></li>
<li class="chapter" data-level="4.4.2" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#reactome"><i class="fa fa-check"></i><b>4.4.2</b> Reactome</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#domain-databases"><i class="fa fa-check"></i><b>4.5</b> Domain databases</a><ul>
<li class="chapter" data-level="4.5.1" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#pfam"><i class="fa fa-check"></i><b>4.5.1</b> Pfam</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#motif-databases"><i class="fa fa-check"></i><b>4.6</b> Motif databases</a></li>
<li class="chapter" data-level="4.7" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#interaction-networks"><i class="fa fa-check"></i><b>4.7</b> Interaction networks</a><ul>
<li class="chapter" data-level="4.7.1" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#biogrid"><i class="fa fa-check"></i><b>4.7.1</b> BioGRID</a></li>
<li class="chapter" data-level="4.7.2" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#string"><i class="fa fa-check"></i><b>4.7.2</b> STRING</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#bioformatics-software"><i class="fa fa-check"></i><b>4.8</b> Bioformatics software</a><ul>
<li class="chapter" data-level="4.8.1" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#blast-1"><i class="fa fa-check"></i><b>4.8.1</b> Blast</a></li>
<li class="chapter" data-level="4.8.2" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#jalview"><i class="fa fa-check"></i><b>4.8.2</b> Jalview</a></li>
<li class="chapter" data-level="4.8.3" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#bioconductor"><i class="fa fa-check"></i><b>4.8.3</b> Bioconductor</a></li>
<li class="chapter" data-level="4.8.4" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#cytoscape"><i class="fa fa-check"></i><b>4.8.4</b> Cytoscape</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="reproducible-research.html"><a href="reproducible-research.html"><i class="fa fa-check"></i><b>5</b> Reproducible research</a><ul>
<li class="chapter" data-level="5.1" data-path="reproducible-research.html"><a href="reproducible-research.html#git"><i class="fa fa-check"></i><b>5.1</b> Git</a><ul>
<li class="chapter" data-level="5.1.1" data-path="reproducible-research.html"><a href="reproducible-research.html#github"><i class="fa fa-check"></i><b>5.1.1</b> Github</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="reproducible-research.html"><a href="reproducible-research.html#literate-programming"><i class="fa fa-check"></i><b>5.2</b> Literate programming</a><ul>
<li class="chapter" data-level="5.2.1" data-path="reproducible-research.html"><a href="reproducible-research.html#rmarkdown-notebooks"><i class="fa fa-check"></i><b>5.2.1</b> RMarkdown notebooks</a></li>
<li class="chapter" data-level="5.2.2" data-path="reproducible-research.html"><a href="reproducible-research.html#jupyter-notebooks"><i class="fa fa-check"></i><b>5.2.2</b> Jupyter notebooks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html"><i class="fa fa-check"></i><b>6</b> Learning Bioinformatics</a><ul>
<li class="chapter" data-level="6.1" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#books"><i class="fa fa-check"></i><b>6.1</b> Books</a><ul>
<li class="chapter" data-level="6.1.1" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#sequence-analaysis"><i class="fa fa-check"></i><b>6.1.1</b> Sequence analaysis</a></li>
<li class="chapter" data-level="6.1.2" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#omics-technologies"><i class="fa fa-check"></i><b>6.1.2</b> Omics technologies</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#community-resources"><i class="fa fa-check"></i><b>6.2</b> Community resources</a><ul>
<li class="chapter" data-level="6.2.1" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#biostars"><i class="fa fa-check"></i><b>6.2.1</b> Biostars</a></li>
<li class="chapter" data-level="6.2.2" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#bioinformatics-channel-at-stackexchange"><i class="fa fa-check"></i><b>6.2.2</b> Bioinformatics channel at Stackexchange</a></li>
<li class="chapter" data-level="6.2.3" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#bioconductor-support-site"><i class="fa fa-check"></i><b>6.2.3</b> Bioconductor support site</a></li>
<li class="chapter" data-level="6.2.4" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#how-to-use-the-support-sites"><i class="fa fa-check"></i><b>6.2.4</b> How to use the support sites</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to Bioinformatics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="omics-analysis-1" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> Omics analysis</h1>
<p>This chapter gives a brief introduction on the most common omics technologies.</p>
<div id="introduction" class="section level2">
<h2><span class="header-section-number">3.1</span> Introduction</h2>
<p>What is omics? In the last 20 years there has been an explosion of words ending in ~ome and ~omics in the scientific literature. All these words probably derive from the word genome and genomics. The definition for genome in a classic textbook in Biology says: “Genome is the complete set of genes of an organism”. Genomics, would be the science that studies the genome.</p>
<p>In a similar way, all the RNA molecules expressed in a cell at a particular time is called transcriptome. For all the proteins expressed in a cell we use proteome. For all the metabolites we use metabolome, and so on. The ome word is also used to describe the complete set of other “things” that are not molecules but some kind of property, feature or virtually almost anything. For example, interactome may refer to the complete set of interactions between molecules in the cell. Extensions of this terminology have resulted in the appearance of terms like phenome, diseasome, and so on. The technologies used to study “omes” are called “omics”. For example, transcriptomics is the field that studies the transcriptome. Proteomics is the study of the proteome, etc. This leads to a whole world of omics technologies.</p>
<div class="figure"><span id="fig:omics-1"></span>
<img src="pic/omics_1.png" alt="A world of OMICS technologies." width="712" />
<p class="caption">
Figure 3.1: A world of OMICS technologies.
</p>
</div>
<p>Let’s look into the basic idea behind one of these technologies and what kind of information they allow us to obtain. For example, here we have one cell before and after exposure to some stimulus. Transcripts are represented by these ribbons, each color being a different transcript. Imagine we are interested on the effect of some chemical on our cells. We hypothesize that the expression of some genes may be altered by the effect of the stimulus, but we do not know whether this is true or what genes may be altered. We decide to do transcriptomics, or gene expression profiling, to identify which genes are affected. Here we can see the result from such experiment. In this hypothetical scenario gene 1 has two transcripts, but only the first one is downregulated, i.e. its expression is reduced after exposure to the stimulus. Gene 4 has one transcript that is upregulated, i.e. its expression is increased by the stimulus. Other transcripts seem to be unchanged. The expression level for all transcripts at a given condition is called the expression profile. By comparing expression profiles we can reveal which genes are altered between the different conditions. These are called differentially expressed genes. One of the goals of transcriptomics is to identify DEGs between different experimental conditions (for example, between control and treatment, patient and disease, etc.).</p>
<div class="figure"><span id="fig:omics-2"></span>
<img src="pic/omics_2.png" alt="Rationale behind OMICS technologies." width="692" />
<p class="caption">
Figure 3.2: Rationale behind OMICS technologies.
</p>
</div>
<p>Existing omics technologies can be roughly divided into three categories: microarray based, next generation sequencing approaches (or NGS) and mass-spectrometry methods (or MS). We will briefly describe some of these techniques in the following videos.</p>
<div class="figure"><span id="fig:omics-3"></span>
<img src="pic/omics_3.png" alt="Classification of OMICS technologies." width="750" />
<p class="caption">
Figure 3.3: Classification of OMICS technologies.
</p>
</div>
</div>
<div id="microarrays" class="section level2">
<h2><span class="header-section-number">3.2</span> Microarrays</h2>
<div id="expression-microarrays" class="section level3">
<h3><span class="header-section-number">3.2.1</span> Expression microarrays</h3>
<p>Microarrays were arguably the first of the “omics” technologies, starting a new generation of high-throughput analyses. This type of exploratory analysis enables the generation of new hypotheses that can be further tested, becoming one important source of knowledge discovery. Microarrays were the first technology to allow this and currently there is a wealth of experimental data available in public databases. There are over 50,000 samples for expression microarrays deposited in the NCBI gene expression omnibus (Figure <a href="omics-analysis-1.html#fig:geo">3.4</a>).</p>
<div class="figure"><span id="fig:geo"></span>
<img src="pic/geo_total.pdf" alt="Number of total submissions for the top 4 technologies."  />
<p class="caption">
Figure 3.4: Number of total submissions for the top 4 technologies.
</p>
</div>
<p>Microarrays are basically made of a solid surface with short oligonucleotides attached to it (see Figure <a href="omics-analysis-1.html#fig:microarray">3.5</a>). Each oligonucleotide has a specific sequence, complementary to a particular gene transcript. This makes the oligonucleotide a probe for the expression level of the target transcript. The solid surface is divided into regions forming an array. Each region has attached the same oligonucleotide and therefore measures the expression of a single transcript.</p>
<div class="figure"><span id="fig:microarray"></span>
<img src="pic/microarray.png" alt="Microarrays are made of oligonucleotide probes attached to a solid surface." width="674" />
<p class="caption">
Figure 3.5: Microarrays are made of oligonucleotide probes attached to a solid surface.
</p>
</div>
<p>Here we can see a typical microarray experiment protocol in more detail (Figure <a href="omics-analysis-1.html#fig:microarray-workflow">3.6</a>). The RNA extracted from the target cells is converted into cDNA using nucleotides with a fluorescence molecule attached. A hybridization reaction is performed to allow the labelled cDNA to bind to its complementary probes. The location of the hybridized probes is revealed by exciting the array with a laser. Only those regions with cDNA bound will emit fluorescence upon stimulation with the laser. Because the location and sequence of the oligonucleotides are known, the location of the fluorescence will tell us which transcript has been detected. Image analysis is performed to quantify the signal associated with each probe, obtaining an estimate of the transcript’s expression level. Analysis of the raw data involves preprocessing of the intensities (usually background correction and normalization) and statistical analysis. Variants of this basic protocol exist that use different labeling approaches, detection methods, amplification of original RNA, etc.</p>
<div class="figure"><span id="fig:microarray-workflow"></span>
<img src="pic/microarray_workflow.png" alt="Summary of typical microarray workflow." width="722" />
<p class="caption">
Figure 3.6: Summary of typical microarray workflow.
</p>
</div>
</div>
<div id="analysis-of-microarray-data" class="section level3">
<h3><span class="header-section-number">3.2.2</span> Analysis of microarray data</h3>
<p>Once the hybridization of the samples and microarrays is finished, we need to measure the signal. This is done by exciting the microarray with a laser and quantifying the fluerescence light emited. An image of the microarray is then generated.</p>
<p>In the image only the regions of the microarray where the cDNA has hybridized will emit light, so we can identify the probes by considering the position of the emited light. The intensity of the light is proportional to the amount of cDNA in the original sample complementary for each probe. So it becomes an estimate of the expression level.</p>
<p>Typically this quantification process is automated and done by the company or service performing the microarray experiment. So our data corresponds with a file containing information about each of the probes in the array and the raw signal (intensity) of the probe, plus some estimate of the noise associated with each probe. From here, the following steps are necessary:</p>
<ol style="list-style-type: decimal">
<li>Load the data into the analysis software (e.g. into R).</li>
<li>Preprocess the microarray signal.</li>
<li>Background correction.</li>
<li>Normalization.</li>
<li>Summarization (for Affymetrix microarrays).</li>
<li>Statistical analysis.</li>
<li>Writing report sumarizing the results.</li>
</ol>
<p>For example, here we can se the output of some microarray platform. The files contain information about each probe position and the intensities.</p>
<pre><code>[CEL]
Version=3

[HEADER]
Cols=602
Rows=602
TotalX=602
TotalY=602
OffsetX=0
OffsetY=0
GridCornerUL=234 232
GridCornerUR=3844 228
GridCornerLR=3847 3838
GridCornerLL=236 3842
Axis-invertX=0
AxisInvertY=0
swapXY=0
DatHeader=[0..24458]  Madrid pool TX+T3
Algorithm=Percentile
AlgorithmParameters=Percentile:75;CellMargin:2;OutlierHigh:1.500;OutlierLow:1.004

[INTENSITY]
NumberCells=362404
CellHeader=X    Y       MEAN    STDV    NPIXELS
  0       0     114.0   16.4     16
  1       0     8715.8  982.2    16
  2       0     137.3   18.5     16
  3       0     9857.0  1267.9   16
  4       0     65.3    13.3     16
  5       0     121.3   16.6     16
  6       0     8474.8  1411.3   16
  7       0     129.8   27.2     16
  8       0     8462.5  1526.1   16
  9       0     130.8   25.6     16
 10       0     9004.8  1564.8   16
 11       0     120.3   15.3     16
 12       0     9083.0  1590.8   16
 13       0     117.0   21.2     16
 14       0     9177.3  1692.9   16
 15       0     123.0   15.7     16
 16       0     9258.0  1606.6   16
 17       0     116.8   19.3     16
 18       0     9016.0  1662.7   16
 19       0     124.3   22.4     16
</code></pre>
</div>
</div>
<div id="next-generation-sequencing" class="section level2">
<h2><span class="header-section-number">3.3</span> Next-generation sequencing</h2>
<p>An important limitation of microarrays is that they are based on oligonucleotide probes for detecting expression. This means microarrays can only detect molecules for which a probe exists in the array. The probes in a microarray are based on our knowledge of the genome’s sequence at the time of microarray design. Because our knowledge is not perfect, microarrays often contain probes measuring things different to what were originally intended. This may lead to incorrect conclusions if the information about the most likely target sequence is not up to date. Instead of using a probe to quantify the presence of a DNA or RNA molecule, why not just sequence the molecule? The problem was that existing sequencing technology was slow and expensive. This motivated the development of so called next generation technologies, or NGS.</p>
<div id="rna-seq" class="section level3">
<h3><span class="header-section-number">3.3.1</span> RNA-seq</h3>
<p>Let’s look into the fundamental concept behind all NGS technologies from the perspective of transcriptomics (Figure <a href="omics-analysis-1.html#fig:ngs">3.7</a>). Imagine this gene containing one intron and both 5’ and 3’ UTRs, and is transcribed into a messenger RNA. The RNA is extracted from the cell and converted into cDNA. This cDNA is used in microarray technology for the hybridization reaction. In NGS however, the cDNA is ligated to sequencing primers, and sequencing produces a set of short reads. Then, this reads are mapped back using sequence alignment, to some reference genome. The location and number of reads will reveal, after correcting from biases like sequencing depth and transcript length, which genes are being expressed and their abundance levels. There are many applications of NGS technologies. Next we will review two of the most popular.</p>
<div class="figure"><span id="fig:ngs"></span>
<img src="pic/ngs.png" alt="Summary of rationale behind NGS technologies." width="712" />
<p class="caption">
Figure 3.7: Summary of rationale behind NGS technologies.
</p>
</div>
<p>The most popular application of NGS technologies is RNA-seq, a technology to measure the transcriptome (Figure <a href="omics-analysis-1.html#fig:ngs-rnaseq">3.8</a>). This slide shows a summary of the typical RNA-seq protocol.</p>
<ol style="list-style-type: decimal">
<li><p>Isolate and fragment the RNA.</p></li>
<li><p>A sequencing library is generated by converting the fragmented molecules into double stranded cDNA and ligating sequencing primers to the ends.</p></li>
<li><p>The cDNA library is sequenced, resulting in a lot of oligonucleotide sequences called “reads”.</p></li>
<li><p>The reads are aligned to a reference genome. The coverage at each genome location is defined as the number of overlapping reads. This density is proportional to the original molecule’s concentration. Therefore, molecules found at higher concentrations will show larger “peaks”.</p></li>
<li><p>Computational methods are used to quantify the peaks overlapping gene sequences. Statistical methods are used to identify genes with differences in expression level.</p></li>
</ol>
<p>Because we are not limited to pre-specified sequence probes we can use the NGS data to detect the expression of alternative splice variants, non-coding RNAs, genetic variants (or SNPs) and even the expression of uncharacterized transcripts. Also, if the reference genome information is updated, the original data can be realigned, never becoming obsolete and possibly revealing additional information. This versatility increases dramatically the amount of information that can be obtained from a single experiment, making RNA-seq one of the most widely used omics technologies.</p>
<div class="figure"><span id="fig:ngs-rnaseq"></span>
<img src="pic/ngs_rnaseq.png" alt="Summary of typical RNA-seq workflow." width="666" />
<p class="caption">
Figure 3.8: Summary of typical RNA-seq workflow.
</p>
</div>
</div>
<div id="example-stat3-expression-in-macrophages-exposed-to-il10" class="section level3">
<h3><span class="header-section-number">3.3.2</span> Example: Stat3 expression in macrophages exposed to IL10</h3>
<p>In the following example we will look into an RNA-seq experiment in more detail. Figure <a href="omics-analysis-1.html#fig:stat3-rnaseq-1">3.9</a> shows the mouse Stat3 locus including the exon/intron structure of the different transcripts as yellow boxes and lines. You can see also other genes nearby, like Stat5a. The expression level in resting and IL10 stimulated macrophages is shown as the peaks overlapping with the exons.</p>
<div class="figure"><span id="fig:stat3-rnaseq-1"></span>
<img src="pic/stat3_rnaseq_1.png" alt="Expression of Stat3 in macrophages exposed to IL10." width="811" />
<p class="caption">
Figure 3.9: Expression of Stat3 in macrophages exposed to IL10.
</p>
</div>
<p>Those peaks represent the density of reads mapping to the genome (Figure <a href="omics-analysis-1.html#fig:stat3-rnaseq-2">3.10</a>). The expression level can be computed from the area of the peaks. Here we see that the expression level of Stat3 is increased in IL10 treated macrophages. From this picture we can also see some of the difficulties found when interpreting this type of data. For example, which of all the Stat3 transcripts are expressed and which are regulated? Answering this question requires more sophisticated analysis and possibly additional experiments designed specifically to gather supporting evidence.</p>
<div class="figure"><span id="fig:stat3-rnaseq-2"></span>
<img src="pic/stat3_rnaseq_2.png" alt="Expression of Stat3 in macrophages exposed to IL10." width="750" />
<p class="caption">
Figure 3.10: Expression of Stat3 in macrophages exposed to IL10.
</p>
</div>
<p>In Figure <a href="omics-analysis-1.html#fig:stat3-rnaseq-3">3.11</a> we can see more details for one of the samples, with the coverage and the aligned reads below. These dense grey boxes are the actual sequenced reads.</p>
<div class="figure"><span id="fig:stat3-rnaseq-3"></span>
<img src="pic/stat3_rnaseq_3.png" alt="Expression of Stat3 in macrophages exposed to IL10." width="750" />
<p class="caption">
Figure 3.11: Expression of Stat3 in macrophages exposed to IL10.
</p>
</div>
<p>Let’s take a closer look to this region near the first exon (Figure <a href="omics-analysis-1.html#fig:stat3-rnaseq-4">3.12</a>). You can see here this boxes indicating the length of the reads and the arrowed end the strand to which they align. If we take an even closer look we can see now the reads in more detail. From this picture you can verify that the coverage is the number of reads overlapping at each genomic location. Here the grey boxes indicate that the read sequence matches that of the reference genome to which they have been aligned. Some of these reads locations, however, show mismatches. These are probably sequencing errors, but if a location shows a mismatch consistently found in all reads it will suggest a SNP in the assayed specimen compared to the reference genome.</p>
<div class="figure"><span id="fig:stat3-rnaseq-4"></span>
<img src="pic/stat3_rnaseq_4.png" alt="Expression of Stat3 in macrophages exposed to IL10." width="750" />
<p class="caption">
Figure 3.12: Expression of Stat3 in macrophages exposed to IL10.
</p>
</div>
</div>
<div id="chip-seq" class="section level3">
<h3><span class="header-section-number">3.3.3</span> ChIP-seq</h3>
<p>Another popular NGS technology is ChIP-seq, which stands for chromatin immunoprecipitation followed by NGS sequencing. In ChIP-seq we aim to identify protein-DNA interactions. That is, we want to know the sequence of the DNA to which a DNA-binding protein binds. This information is critical to understand gene regulatory programs. For example, gene transcription is regulated by transcription factors that bind to specific regions in the genome, including gene promoters and enhancers. Also, histones represent another type of DNA binding proteins with important roles in the regulation of gene expression. Posttranscriptional histone modifications, like acetylation, methylation, and others, are associated with regulatory outcomes, like expression activation and repression. ChIP-seq enables to study many of these patterns at genome wide scale.</p>
<p>The basic ChIP-seq protocol is shown in Figure <a href="omics-analysis-1.html#fig:ngs-chipseq">3.13</a>. First, proteins are fixed to the DNA with, e.g. formaldehyde. Then, the chromatin is fragmented, leading to many small fragments free of proteins and some with proteins bound. Among the later ones, some will be bound to the target protein, that can be a transcription factor or a histone with some particular modification like H3K9me3 (tri-methylation of lysine in position 9 of the histone H3). Next, we use an antibody against the target protein for an immunoprecipitation, which allows us to obtain the proteins and the DNA sequences they are bound to. Then we remove the fixation and isolate the DNA. The rest of the protocol, cDNA library construction and sequencing are performed similarly to RNA-seq. Sequencing will produce sequence reads that can be aligned to the reference genome. The read peaks accumulating in DNA loci will reveal where the target protein was bound. Computational methods are used to identify and quantify the peaks. Statistical analysis will help us determine if there is differential binding between experimental conditions.</p>
<div class="figure"><span id="fig:ngs-chipseq"></span>
<img src="pic/ngs_chipseq.png" alt="ChIP-seq workflow." width="686" />
<p class="caption">
Figure 3.13: ChIP-seq workflow.
</p>
</div>
</div>
<div id="example-stat3-binding-in-macrophages-exposed-to-il10" class="section level3">
<h3><span class="header-section-number">3.3.4</span> Example: Stat3 binding in macrophages exposed to IL10</h3>
<p>In the following example we can see an application of ChIP-seq to identify the binding locations of the transcription factor Stat3 in macrophages after exposure to Il10. As before, the Stat3 locus in the mouse genome is shown, together with the coverage for the Stat3 ChIP-seq experiment. We can see that Stat3 binds to its own promoter upon Il10 treatment, suggesting that it regulates its own expression.</p>
<div class="figure"><span id="fig:stat3-chipseq"></span>
<img src="pic/stat3_chipseq.png" alt="Stat3 binding profile around the Stat3 locus" width="836" />
<p class="caption">
Figure 3.14: Stat3 binding profile around the Stat3 locus
</p>
</div>
<p>An important limitation of ChIP-seq is that it can only identify binding sites for a single TF or histone modification at a time. Another problem is that in eukaryotes, enhancers can be located far away from the genes they regulate, hampering the assignment of TF binding to target genes. Finally, the observation of protein binding to DNA does not demonstrate its functional relevance, which requires additional functional experiments.</p>
</div>
<div id="other-ngs-technologies" class="section level3">
<h3><span class="header-section-number">3.3.5</span> Other NGS technologies</h3>
<p>Other NGS technologies are becoming popular in recent years. For example, TSS-seq and CAGE-seq are technologies that identify the 5’ cap structure of the processed mRNA, revealing the location of active promoters. Other technologies are able to reveal information about the chromatin. Chromatin accessibility can be assayed by means of either DNaseI digestion or by transposase reaction. In both cases the enzymes have a preference to react with open chromatin regions, with regions of close chromatin or protected by the binding of proteins being more resilient. This enables to map open chromatin and even the location of nucleosomes and binding of TFs if enough sequencing depth is performed. Nucleosome location can be more directly assayed by using the Mnase-seq protocol. Finally, the methylation of the DNA can be assayed by bisulfite sequencing.</p>
</div>
</div>
<div id="mass-spectromics-methods" class="section level2">
<h2><span class="header-section-number">3.4</span> Mass spectromics methods</h2>
<p>Studying transcriptional regulation is one important part for understanding the cell’s phenotype. But some of the transcribed genes will produce proteins that engage in diverse functions in the cell, including signaling pathways that regulate the cell’s behavior and enzymes that regulate metabolic pathways. Therefore, the other important part to understand the cell’s phenotype are the proteome and metabolome. For studying the proteome mass-spectrometry (or MS) approaches have been fundamental. Here we will focus on the use of MS approaches for proteomics.</p>
<p>MS uses the mass-to-charge ratio to identify and quantify chemical molecules. MS is used not only to quantify protein expression, but also post-translational modifications like phosphorylation or ubiquitination, and protein-protein interactions.</p>
<div id="ms" class="section level3">
<h3><span class="header-section-number">3.4.1</span> MS</h3>
<p>The general proteomics protocol requires digesting the proteins into fragments (peptides). These will be initially separated by liquid chromatography (LC) coupled to MS. This will lead to peptide separation but identification requires another round of fragmentation followed by MS. Each peptide produces a distinct spectrum, which is compared against a database linking spectra to peptide sequences. The identified peptides are then mapped to a protein database to identify the proteins they came from. The number of mapping peptides will be proportional to the original abundance of the proteins, enabling their quantification.</p>
<div class="figure"><span id="fig:ms"></span>
<img src="pic/ms.png" alt="Workflow for MS proteomics." width="586" />
<p class="caption">
Figure 3.15: Workflow for MS proteomics.
</p>
</div>
</div>
<div id="ap-ms" class="section level3">
<h3><span class="header-section-number">3.4.2</span> AP-MS</h3>
<p>Protein interactions are fundamental in regulating cellular pathways, and they play as well important roles in immunology. To obtain protein-protein interaction information we can use a modified MS protocol called AP-MS (affinity purification followed by MS). Similar to ChIP-seq, the first step is to crosslink interacting proteins together with formaldehyde. Next, we isolate the complexes with an antibody against the target protein, also call the bait. Then, crosslinking is reversed, the proteins are digested and LC-MS/MS performed in the usual way. Mapping the peptides to the proteome will identify which other proteins where interacting with our target protein. AP-MS can only assay the interactome around the target protein. Also, it does not identify direct interactions, but proteins that are part of the same protein complex.</p>
<div class="figure"><span id="fig:ap-ms"></span>
<img src="pic/ap-ms.png" alt="Workflow for AP-MS proteomics for identifying protein-protein interactions." width="818" />
<p class="caption">
Figure 3.16: Workflow for AP-MS proteomics for identifying protein-protein interactions.
</p>
</div>
</div>
</div>
<div id="single-cell-omics" class="section level2">
<h2><span class="header-section-number">3.5</span> Single cell omics</h2>
<p>So far we have assumed that in each omic experiment we obtain a tissue sample, or perhaps isolate the cell type of interest with FACS and then perform the analysis on the bulk of cells. An experiment performed in this way reveals information about (in the case of transcriptomics) the average expression level of all cells in the bulk. In general, this approach is valid if we believe that most of the cells in the population are similarly and that they behavior is homogeneous. However, recent evidence shows that even in seemingly homogenous populations, there is substantial heterogeneity in the expression levels of genes and proteins. This is even more dramatic in populations where a few cells are regulated distinctly and initiate changes that will drive future responses. For example, in the lymph node there are many naive T cells but only the ones in contact with their specific antigen will be activated and mature into effector cells. Because the associated changes occur in a very limited number of cells, that information may be lost in the averaging resulting from analyzing bulks of cells. This is the motivation behind the field of single cells omics. The foundation of single cell omics is almost identical to what we have seen so far. The main difference is that these omics technologies are applied to each cell individually, requiring of methods to manipulate and extract the target material from single cells in parallel and efficiently. Another challenge is the analysis of this type of data. The data resulting from single cell omics is multivariate in nature and specialized methods are required for their interpretation.</p>
<div class="figure"><span id="fig:sc"></span>
<img src="pic/ngs_sc.png" alt="Single cell omics captures the stochastic nature of biological processes" width="548" />
<p class="caption">
Figure 3.17: Single cell omics captures the stochastic nature of biological processes
</p>
</div>
<p>The field of single cell omics is in constant development and new or improved methodologies appear increasingly. For transcriptomics the use of microfluidic reaction chambers enables single cells to be subjected to RNA-seq protocols. Another methodology is Fluorescence based flow cytometry (FBFC), which enables to quantify the amount of proteins using fluorophores attached to antibodies. However, because spectral overlap, the maximum number of proteins that can be measured simultaneously is limited. A modification of this technology known as mass cytometry tags antibodies with rare isotopes and uses mass spectrometry to identify proteins. This technique highly improves the dimensionality by increasing the number of proteins and conditions that can be analyzed simultaneously.</p>
</div>
<div id="batch-effects" class="section level2">
<h2><span class="header-section-number">3.6</span> Batch effects</h2>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="sequence-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="bioinformatics-resources.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["nagoya_course_book.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
