<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Introduction to Bioinformatics</title>
  <meta name="description" content="Material covered in the <em>Introduction to Bioinformatics</em> intensive course held in Nagoya University, Dec 13-14, 2017.">
  <meta name="generator" content="bookdown 0.5 and GitBook 2.6.7">

  <meta property="og:title" content="Introduction to Bioinformatics" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Material covered in the <em>Introduction to Bioinformatics</em> intensive course held in Nagoya University, Dec 13-14, 2017." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Introduction to Bioinformatics" />
  
  <meta name="twitter:description" content="Material covered in the <em>Introduction to Bioinformatics</em> intensive course held in Nagoya University, Dec 13-14, 2017." />
  

<meta name="author" content="Diego Diez">


<meta name="date" content="2017-12-05">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="intro.html">
<link rel="next" href="omics-analysis-1.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Introduction to Bioinformatics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About</a></li>
<li class="chapter" data-level="" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i>Prerequisites</a></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="intro.html"><a href="intro.html#sequence-analysis"><i class="fa fa-check"></i><b>1.1</b> Sequence analysis</a></li>
<li class="chapter" data-level="1.2" data-path="intro.html"><a href="intro.html#omics-analysis"><i class="fa fa-check"></i><b>1.2</b> Omics analysis</a></li>
<li class="chapter" data-level="1.3" data-path="intro.html"><a href="intro.html#structural-analysis"><i class="fa fa-check"></i><b>1.3</b> Structural analysis</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="sequence-analysis-1.html"><a href="sequence-analysis-1.html"><i class="fa fa-check"></i><b>2</b> Sequence analysis</a><ul>
<li class="chapter" data-level="2.1" data-path="sequence-analysis-1.html"><a href="sequence-analysis-1.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="sequence-analysis-1.html"><a href="sequence-analysis-1.html#two-sequence-alignment"><i class="fa fa-check"></i><b>2.2</b> Two sequence alignment</a><ul>
<li class="chapter" data-level="2.2.1" data-path="sequence-analysis-1.html"><a href="sequence-analysis-1.html#comparing-two-sequences-of-different-lengths"><i class="fa fa-check"></i><b>2.2.1</b> Comparing two sequences of different lengths</a></li>
<li class="chapter" data-level="2.2.2" data-path="sequence-analysis-1.html"><a href="sequence-analysis-1.html#smith-waterman-global-alignment"><i class="fa fa-check"></i><b>2.2.2</b> Smith-waterman (global alignment)</a></li>
<li class="chapter" data-level="2.2.3" data-path="sequence-analysis-1.html"><a href="sequence-analysis-1.html#needleman--local-alignment"><i class="fa fa-check"></i><b>2.2.3</b> Needleman- (local alignment)</a></li>
<li class="chapter" data-level="2.2.4" data-path="sequence-analysis-1.html"><a href="sequence-analysis-1.html#blast"><i class="fa fa-check"></i><b>2.2.4</b> Blast</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="sequence-analysis-1.html"><a href="sequence-analysis-1.html#multiple-sequence-alignment"><i class="fa fa-check"></i><b>2.3</b> Multiple sequence alignment</a></li>
<li class="chapter" data-level="2.4" data-path="sequence-analysis-1.html"><a href="sequence-analysis-1.html#domains-and-motifs"><i class="fa fa-check"></i><b>2.4</b> Domains and motifs</a></li>
<li class="chapter" data-level="2.5" data-path="sequence-analysis-1.html"><a href="sequence-analysis-1.html#phylogenetic-methods"><i class="fa fa-check"></i><b>2.5</b> Phylogenetic methods</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html"><i class="fa fa-check"></i><b>3</b> Omics analysis</a><ul>
<li class="chapter" data-level="3.1" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#introduction-1"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#microarrays"><i class="fa fa-check"></i><b>3.2</b> Microarrays</a></li>
<li class="chapter" data-level="3.3" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#next-generation-sequencing"><i class="fa fa-check"></i><b>3.3</b> Next-generation sequencing</a><ul>
<li class="chapter" data-level="3.3.1" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#rna-seq"><i class="fa fa-check"></i><b>3.3.1</b> RNA-seq</a></li>
<li class="chapter" data-level="3.3.2" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#example-stat3-expression-in-macrophages-exposed-to-il10"><i class="fa fa-check"></i><b>3.3.2</b> Example: Stat3 expression in macrophages exposed to IL10</a></li>
<li class="chapter" data-level="3.3.3" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#chip-seq"><i class="fa fa-check"></i><b>3.3.3</b> ChIP-seq</a></li>
<li class="chapter" data-level="3.3.4" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#example-stat3-binding-in-macrophages-exposed-to-il10"><i class="fa fa-check"></i><b>3.3.4</b> Example: Stat3 binding in macrophages exposed to IL10</a></li>
<li class="chapter" data-level="3.3.5" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#other-ngs-technologies"><i class="fa fa-check"></i><b>3.3.5</b> Other NGS technologies</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#mass-spectromics-methods"><i class="fa fa-check"></i><b>3.4</b> Mass spectromics methods</a></li>
<li class="chapter" data-level="3.5" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#single-cell-omics"><i class="fa fa-check"></i><b>3.5</b> Single cell omics</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html"><i class="fa fa-check"></i><b>4</b> Bioinformatics resources</a><ul>
<li class="chapter" data-level="4.1" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#gene-ontology"><i class="fa fa-check"></i><b>4.1</b> Gene Ontology</a></li>
<li class="chapter" data-level="4.2" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#pathway-databases"><i class="fa fa-check"></i><b>4.2</b> Pathway databases</a><ul>
<li class="chapter" data-level="4.2.1" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#kegg"><i class="fa fa-check"></i><b>4.2.1</b> KEGG</a></li>
<li class="chapter" data-level="4.2.2" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#reactome"><i class="fa fa-check"></i><b>4.2.2</b> Reactome</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#domain-databases"><i class="fa fa-check"></i><b>4.3</b> Domain databases</a><ul>
<li class="chapter" data-level="4.3.1" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#pfam"><i class="fa fa-check"></i><b>4.3.1</b> PFAM</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#motif-databases"><i class="fa fa-check"></i><b>4.4</b> Motif databases</a></li>
<li class="chapter" data-level="4.5" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#interaction-networks"><i class="fa fa-check"></i><b>4.5</b> Interaction networks</a><ul>
<li class="chapter" data-level="4.5.1" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#biogrid"><i class="fa fa-check"></i><b>4.5.1</b> Biogrid</a></li>
<li class="chapter" data-level="4.5.2" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#string"><i class="fa fa-check"></i><b>4.5.2</b> STRING</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="reproducible-research.html"><a href="reproducible-research.html"><i class="fa fa-check"></i><b>5</b> Reproducible research</a><ul>
<li class="chapter" data-level="5.1" data-path="reproducible-research.html"><a href="reproducible-research.html#git"><i class="fa fa-check"></i><b>5.1</b> Git</a><ul>
<li class="chapter" data-level="5.1.1" data-path="reproducible-research.html"><a href="reproducible-research.html#github"><i class="fa fa-check"></i><b>5.1.1</b> Github</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="reproducible-research.html"><a href="reproducible-research.html#literate-programming"><i class="fa fa-check"></i><b>5.2</b> Literate programming</a><ul>
<li class="chapter" data-level="5.2.1" data-path="reproducible-research.html"><a href="reproducible-research.html#rmarkdown-notebooks"><i class="fa fa-check"></i><b>5.2.1</b> RMarkdown notebooks</a></li>
<li class="chapter" data-level="5.2.2" data-path="reproducible-research.html"><a href="reproducible-research.html#jupyter-notebooks"><i class="fa fa-check"></i><b>5.2.2</b> Jupyter notebooks</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to Bioinformatics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="sequence-analysis-1" class="section level1">
<h1><span class="header-section-number">Chapter 2</span> Sequence analysis</h1>
<p>This chapter introduces the basics of sequence analysis methods.</p>
<div id="introduction" class="section level2">
<h2><span class="header-section-number">2.1</span> Introduction</h2>
</div>
<div id="two-sequence-alignment" class="section level2">
<h2><span class="header-section-number">2.2</span> Two sequence alignment</h2>
<p>When we want to compare two sequences of the same size a naive method would be line them up and count the number of mismatches.</p>
<pre><code>AAGGATGA
AACGATAA
..*...*.</code></pre>
<div id="comparing-two-sequences-of-different-lengths" class="section level3">
<h3><span class="header-section-number">2.2.1</span> Comparing two sequences of different lengths</h3>
</div>
<div id="smith-waterman-global-alignment" class="section level3">
<h3><span class="header-section-number">2.2.2</span> Smith-waterman (global alignment)</h3>
</div>
<div id="needleman--local-alignment" class="section level3">
<h3><span class="header-section-number">2.2.3</span> Needleman- (local alignment)</h3>
</div>
<div id="blast" class="section level3">
<h3><span class="header-section-number">2.2.4</span> Blast</h3>
</div>
</div>
<div id="multiple-sequence-alignment" class="section level2">
<h2><span class="header-section-number">2.3</span> Multiple sequence alignment</h2>
<p>When we want to align 3 o more sequences together we have a multiple sequence alignment (MSA). In MSA there is more ambiguity in what determines a correct alignment.</p>
<p>Software for MSA:</p>
<ul>
<li>MAFFT</li>
<li>Clustal (Classic)</li>
<li>MUSCLE</li>
</ul>
</div>
<div id="domains-and-motifs" class="section level2">
<h2><span class="header-section-number">2.4</span> Domains and motifs</h2>
<p>Biological sequences contain regions of high similarity to other similar regions in other proteins. Sometimes this regios are small, accounting for a few aminoacids, and they are typically named <em>motifs</em>. In other cases the conserved regions are larger, and then are called <em>domains</em>. Domains are sometimes associated with some structurally motivated definition of domain. Sometimes however, the distinction between these two types of conserved regions is ambigous.</p>
<p>Motifs and domains can be identified by using the information from a MSA. This is the approach followed in the Pfam database. Sections of MSAs containing the domain of interest are manually curated. These are used to train a Hidden Markov Model (HMM) that learns how to recognize the same domain in other sequences. With that model, we can identify all sequences in a database containing that particular domain.</p>
<p>The software HMMER implements HMM for the analysis of biological sequences. It can learn models representing motifs, domains or entire sequences. These models are used to search for sequences matching the models in databases. It can also be used to annotate the domains/motifs present in a sequence using a database of domains. PFAM uses HMMER and manually curated MSA of domains to generate a database of protein families (PFAM).</p>
</div>
<div id="phylogenetic-methods" class="section level2">
<h2><span class="header-section-number">2.5</span> Phylogenetic methods</h2>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="intro.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="omics-analysis-1.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["nagoya_course_book.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
