<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Introduction to Bioinformatics</title>
  <meta name="description" content="Material covered in the <em>Introduction to Bioinformatics</em> intensive course held in Nagoya University, Dec 13-14, 2017.">
  <meta name="generator" content="bookdown 0.5 and GitBook 2.6.7">

  <meta property="og:title" content="Introduction to Bioinformatics" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Material covered in the <em>Introduction to Bioinformatics</em> intensive course held in Nagoya University, Dec 13-14, 2017." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Introduction to Bioinformatics" />
  
  <meta name="twitter:description" content="Material covered in the <em>Introduction to Bioinformatics</em> intensive course held in Nagoya University, Dec 13-14, 2017." />
  

<meta name="author" content="Diego Diez">


<meta name="date" content="2017-12-11">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="omics-analysis-1.html">
<link rel="next" href="reproducible-research.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Introduction to Bioinformatics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About</a></li>
<li class="chapter" data-level="" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i>Prerequisites</a></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="intro.html"><a href="intro.html#intro-sequence-analysis"><i class="fa fa-check"></i><b>1.1</b> Sequence analysis</a></li>
<li class="chapter" data-level="1.2" data-path="intro.html"><a href="intro.html#omics-analysis"><i class="fa fa-check"></i><b>1.2</b> Omics analysis</a></li>
<li class="chapter" data-level="1.3" data-path="intro.html"><a href="intro.html#structural-analysis"><i class="fa fa-check"></i><b>1.3</b> Structural analysis</a></li>
<li class="chapter" data-level="1.4" data-path="intro.html"><a href="intro.html#phylogenetic-analysis"><i class="fa fa-check"></i><b>1.4</b> Phylogenetic analysis</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="sequence-analysis.html"><a href="sequence-analysis.html"><i class="fa fa-check"></i><b>2</b> Sequence analysis</a><ul>
<li class="chapter" data-level="2.1" data-path="sequence-analysis.html"><a href="sequence-analysis.html#pairwise-alignment"><i class="fa fa-check"></i><b>2.1</b> Pairwise alignment</a></li>
<li class="chapter" data-level="2.2" data-path="sequence-analysis.html"><a href="sequence-analysis.html#scoring-matrices"><i class="fa fa-check"></i><b>2.2</b> Scoring matrices</a></li>
<li class="chapter" data-level="2.3" data-path="sequence-analysis.html"><a href="sequence-analysis.html#blast"><i class="fa fa-check"></i><b>2.3</b> Blast</a></li>
<li class="chapter" data-level="2.4" data-path="sequence-analysis.html"><a href="sequence-analysis.html#multiple-sequence-alignment"><i class="fa fa-check"></i><b>2.4</b> Multiple sequence alignment</a></li>
<li class="chapter" data-level="2.5" data-path="sequence-analysis.html"><a href="sequence-analysis.html#tree-methods"><i class="fa fa-check"></i><b>2.5</b> Tree methods</a></li>
<li class="chapter" data-level="2.6" data-path="sequence-analysis.html"><a href="sequence-analysis.html#domains-and-motifs"><i class="fa fa-check"></i><b>2.6</b> Domains and motifs</a></li>
<li class="chapter" data-level="2.7" data-path="sequence-analysis.html"><a href="sequence-analysis.html#phylogenetics"><i class="fa fa-check"></i><b>2.7</b> Phylogenetics</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html"><i class="fa fa-check"></i><b>3</b> Omics analysis</a><ul>
<li class="chapter" data-level="3.1" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#microarrays"><i class="fa fa-check"></i><b>3.2</b> Microarrays</a></li>
<li class="chapter" data-level="3.3" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#next-generation-sequencing"><i class="fa fa-check"></i><b>3.3</b> Next-generation sequencing</a><ul>
<li class="chapter" data-level="3.3.1" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#rna-seq"><i class="fa fa-check"></i><b>3.3.1</b> RNA-seq</a></li>
<li class="chapter" data-level="3.3.2" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#chip-seq"><i class="fa fa-check"></i><b>3.3.2</b> ChIP-seq</a></li>
<li class="chapter" data-level="3.3.3" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#other-ngs-technologies"><i class="fa fa-check"></i><b>3.3.3</b> Other NGS technologies</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#mass-spectromics-methods"><i class="fa fa-check"></i><b>3.4</b> Mass spectromics methods</a><ul>
<li class="chapter" data-level="3.4.1" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#ms"><i class="fa fa-check"></i><b>3.4.1</b> MS</a></li>
<li class="chapter" data-level="3.4.2" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#ap-ms"><i class="fa fa-check"></i><b>3.4.2</b> AP-MS</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#single-cell-omics"><i class="fa fa-check"></i><b>3.5</b> Single cell omics</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html"><i class="fa fa-check"></i><b>4</b> Bioinformatics resources</a><ul>
<li class="chapter" data-level="4.1" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#sequence-databases"><i class="fa fa-check"></i><b>4.1</b> Sequence databases</a><ul>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#ncbi"><i class="fa fa-check"></i>NCBI</a></li>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#ensembl"><i class="fa fa-check"></i>EnsEMBL</a></li>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#uniprot"><i class="fa fa-check"></i>Uniprot</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#gene-expression-databases"><i class="fa fa-check"></i><b>4.2</b> Gene expression databases</a><ul>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#geo-gene-expression-omnibus"><i class="fa fa-check"></i>GEO: Gene Expression Omnibus</a></li>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#arrayexpress"><i class="fa fa-check"></i>ArrayExpress</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#pathway-databases"><i class="fa fa-check"></i><b>4.3</b> Pathway databases</a><ul>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#kegg"><i class="fa fa-check"></i>KEGG</a></li>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#reactome"><i class="fa fa-check"></i>Reactome</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#domain-databases"><i class="fa fa-check"></i><b>4.4</b> Domain databases</a><ul>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#pfam"><i class="fa fa-check"></i>Pfam</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#motif-databases"><i class="fa fa-check"></i><b>4.5</b> Motif databases</a><ul>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#jaspar"><i class="fa fa-check"></i>Jaspar</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#interaction-databases"><i class="fa fa-check"></i><b>4.6</b> Interaction databases</a><ul>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#biogrid"><i class="fa fa-check"></i>BioGRID</a></li>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#string"><i class="fa fa-check"></i>STRING</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#other"><i class="fa fa-check"></i><b>4.7</b> Other</a><ul>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#gene-ontology"><i class="fa fa-check"></i>Gene Ontology</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#bioformatics-software"><i class="fa fa-check"></i><b>4.8</b> Bioformatics software</a><ul>
<li class="chapter" data-level="4.8.1" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#sequence-analysis-1"><i class="fa fa-check"></i><b>4.8.1</b> Sequence analysis</a></li>
<li class="chapter" data-level="4.8.2" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#next-generation-sequencing-1"><i class="fa fa-check"></i><b>4.8.2</b> Next generation sequencing</a></li>
<li class="chapter" data-level="4.8.3" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#general-frameworks"><i class="fa fa-check"></i><b>4.8.3</b> General frameworks</a></li>
</ul></li>
<li class="chapter" data-level="4.9" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#file-formats"><i class="fa fa-check"></i><b>4.9</b> File formats</a><ul>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#fasta"><i class="fa fa-check"></i>FASTA</a></li>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#fastq"><i class="fa fa-check"></i>FASTQ</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="reproducible-research.html"><a href="reproducible-research.html"><i class="fa fa-check"></i><b>5</b> Reproducible research</a><ul>
<li class="chapter" data-level="5.1" data-path="reproducible-research.html"><a href="reproducible-research.html#git"><i class="fa fa-check"></i><b>5.1</b> Git</a><ul>
<li class="chapter" data-level="" data-path="reproducible-research.html"><a href="reproducible-research.html#github"><i class="fa fa-check"></i>Github</a></li>
<li class="chapter" data-level="" data-path="reproducible-research.html"><a href="reproducible-research.html#gitlab"><i class="fa fa-check"></i>Gitlab</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="reproducible-research.html"><a href="reproducible-research.html#literate-programming"><i class="fa fa-check"></i><b>5.2</b> Literate programming</a><ul>
<li class="chapter" data-level="" data-path="reproducible-research.html"><a href="reproducible-research.html#rmarkdown-notebooks"><i class="fa fa-check"></i>RMarkdown notebooks</a></li>
<li class="chapter" data-level="" data-path="reproducible-research.html"><a href="reproducible-research.html#jupyter-notebooks"><i class="fa fa-check"></i>Jupyter notebooks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html"><i class="fa fa-check"></i><b>6</b> Learning Bioinformatics</a><ul>
<li class="chapter" data-level="6.1" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#books"><i class="fa fa-check"></i><b>6.1</b> Books</a><ul>
<li class="chapter" data-level="6.1.1" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#sequence-analysis-2"><i class="fa fa-check"></i><b>6.1.1</b> Sequence analysis</a></li>
<li class="chapter" data-level="6.1.2" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#omics-technologies"><i class="fa fa-check"></i><b>6.1.2</b> Omics technologies</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#online-resources"><i class="fa fa-check"></i><b>6.2</b> Online resources</a><ul>
<li class="chapter" data-level="6.2.1" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#learning-how-to-use-the-command-line"><i class="fa fa-check"></i><b>6.2.1</b> Learning how to use the command line</a></li>
<li class="chapter" data-level="6.2.2" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#support-communities"><i class="fa fa-check"></i><b>6.2.2</b> Support Communities</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to Bioinformatics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="bioinformatics-resources" class="section level1">
<h1><span class="header-section-number">Chapter 4</span> Bioinformatics resources</h1>
<p>In this chapter some of the most important Bioinformatics resources are described. An attempt to organized them by topic is made, but most of these resources are highly integrated. So, for example, you would thing that to do a Blast search you need to go to the Blast web page. But you can also perform Blast searches from the Uniprot and EnsEMBL web pages, to mention just two. Similarly, most if not all sequence databases link the associated Gene Ontology information to their sequence entries. This high degree of integration might make difficult at first to value the existence of different resources.</p>
<div id="sequence-databases" class="section level2">
<h2><span class="header-section-number">4.1</span> Sequence databases</h2>
<div id="ncbi" class="section level3 unnumbered">
<h3>NCBI</h3>
</div>
<div id="ensembl" class="section level3 unnumbered">
<h3>EnsEMBL</h3>
<p><a href="https://www.ensembl.org">EnsEMBL</a> is a database where the genome information about many species is collected with a focus con comparative genomics research.</p>
<div class="figure"><span id="fig:ensembl-web"></span>
<img src="pic/ensembl_web.png" alt="Screenshot of the EnsEMBL home page at https://www.ensembl.org." width="614" />
<p class="caption">
Figure 4.1: Screenshot of the EnsEMBL home page at <a href="https://www.ensembl.org" class="uri">https://www.ensembl.org</a>.
</p>
</div>
</div>
<div id="uniprot" class="section level3 unnumbered">
<h3>Uniprot</h3>
<p>The <a href="http://www.uniprot.org">Uniprot</a> website focuses on information about protein sequences, including splice forms.</p>
<div class="figure"><span id="fig:uniprot-web"></span>
<img src="pic/uniprot_web.png" alt="Screenshot of the Uniprot home page at http://www.uniprot.org." width="610" />
<p class="caption">
Figure 4.2: Screenshot of the Uniprot home page at <a href="http://www.uniprot.org" class="uri">http://www.uniprot.org</a>.
</p>
</div>
</div>
</div>
<div id="gene-expression-databases" class="section level2">
<h2><span class="header-section-number">4.2</span> Gene expression databases</h2>
<div id="geo-gene-expression-omnibus" class="section level3 unnumbered">
<h3>GEO: Gene Expression Omnibus</h3>
<p>The <a href="https://www.ncbi.nlm.nih.gov/geo/">Gene Expression Omnibus (GEO)</a> database collects data from high-throughtput experiments including microarrays and NGS datasets.</p>
<div class="figure"><span id="fig:geo-web"></span>
<img src="pic/geo_web.png" alt="Screenshot of the GEO home page at https://www.ncbi.nlm.nih.gov/geo/." width="590" />
<p class="caption">
Figure 4.3: Screenshot of the GEO home page at <a href="https://www.ncbi.nlm.nih.gov/geo/" class="uri">https://www.ncbi.nlm.nih.gov/geo/</a>.
</p>
</div>
<p>The number of submissions per year for microarrays and RNA-seq continues to increase.</p>
<div class="figure"><span id="fig:geo-year"></span>
<img src="pic/geo_year.pdf" alt="Number of GEO submissions per year for the top 4 technologies."  />
<p class="caption">
Figure 4.4: Number of GEO submissions per year for the top 4 technologies.
</p>
</div>
</div>
<div id="arrayexpress" class="section level3 unnumbered">
<h3>ArrayExpress</h3>
<p><a href="https://www.ebi.ac.uk/arrayexpress/">ArrayExpress</a> is the alternative at the European Bioinformatics Institute.</p>
<div class="figure"><span id="fig:ae-web"></span>
<img src="pic/ae_web.png" alt="Screenshot of the ArrayExpress home page at https://www.ebi.ac.uk/arrayexpress/" width="594" />
<p class="caption">
Figure 4.5: Screenshot of the ArrayExpress home page at <a href="https://www.ebi.ac.uk/arrayexpress/" class="uri">https://www.ebi.ac.uk/arrayexpress/</a>
</p>
</div>
</div>
</div>
<div id="pathway-databases" class="section level2">
<h2><span class="header-section-number">4.3</span> Pathway databases</h2>
<p>As information about genes, proteins and metabolites accumulates, it becomes more and more important to organize this knowledge in a comprehensive way. We have mentioned how genes and their products participate in the networks of cellular processes, the cell’s pathway. This network includes metabolic, signaling and gene regulatory networks. This information is critical to understand the cell’s phenotype, but the accumulated knowledge is daunting. It would benefit to have some central repository with easily accessible tools that enables to visualize existing knowledge in an easy way. It would be also very useful if it was possible to use that information in computer analysis. This is the motivation behind some of the web resources described below.</p>
<div id="kegg" class="section level3 unnumbered">
<h3>KEGG</h3>
<p>KEGG stands for <a href="http://www.kegg.jp">Kyoto Encyclopedia for Genes and Genomes</a> (Figure <a href="bioinformatics-resources.html#fig:kegg-web">4.6</a>). Is an integrative resource with information about matabolic and signaling pathways in different species, from bacteria and archaea to mammals.</p>
<div class="figure"><span id="fig:kegg-web"></span>
<img src="pic/kegg_web.png" alt="Screenshot of the KEGG home page at http://www.kegg.jp." width="374" />
<p class="caption">
Figure 4.6: Screenshot of the KEGG home page at <a href="http://www.kegg.jp" class="uri">http://www.kegg.jp</a>.
</p>
</div>
</div>
<div id="reactome" class="section level3 unnumbered">
<h3>Reactome</h3>
</div>
</div>
<div id="domain-databases" class="section level2">
<h2><span class="header-section-number">4.4</span> Domain databases</h2>
<div id="pfam" class="section level3 unnumbered">
<h3>Pfam</h3>
<p><a href="http://pfam.xfam.org">Pfam</a> is a database of protein domain families.</p>
<blockquote>
<p>Proteins are generally comprised of one or more functional regions, commonly termed domains. The presence of different domains in varying combinations in different proteins gives rise to the diverse repertoire of proteins found in nature. Identifying the domains present in a protein can provide insights into the function of that protein.</p>
<p>The Pfam database is a large collection of protein domain families. Each family is represented by multiple sequence alignments and a hidden Markov model (HMMs).</p>
</blockquote>
<div class="figure"><span id="fig:pfam-web"></span>
<img src="pic/pfam_web.png" alt="Screenshot of the Pfam home page at http://pfam.xfam.org." width="611" />
<p class="caption">
Figure 4.7: Screenshot of the Pfam home page at <a href="http://pfam.xfam.org" class="uri">http://pfam.xfam.org</a>.
</p>
</div>
</div>
</div>
<div id="motif-databases" class="section level2">
<h2><span class="header-section-number">4.5</span> Motif databases</h2>
<div id="jaspar" class="section level3 unnumbered">
<h3>Jaspar</h3>
<p><a href="http://jaspar.genereg.net">Jaspar</a> is a database for transcription factor DNA binding motifs.</p>
<div class="figure"><span id="fig:jaspar-web"></span>
<img src="pic/jaspar_web.png" alt="Screenshot of the Jaspar home page at http://jaspar.genereg.net." width="613" />
<p class="caption">
Figure 4.8: Screenshot of the Jaspar home page at <a href="http://jaspar.genereg.net" class="uri">http://jaspar.genereg.net</a>.
</p>
</div>
</div>
</div>
<div id="interaction-databases" class="section level2">
<h2><span class="header-section-number">4.6</span> Interaction databases</h2>
<p>Information about protein-protein interactions has grown incredibly due to advances in MS in combination with affinity purification methods. There are several databases integrating existing information. Two resources will be highlighted here: BioGRID and STRING.</p>
<div id="biogrid" class="section level3 unnumbered">
<h3>BioGRID</h3>
<p><a href="https://thebiogrid.org">BioGRID</a> focuses on storing information about PPIs for which experimental evidence exists.</p>
<div class="figure"><span id="fig:biogrid-web"></span>
<img src="pic/biogrid_web.png" alt="Screenshot of the BioGRID home page at https://thebiogrid.org." width="510" />
<p class="caption">
Figure 4.9: Screenshot of the BioGRID home page at <a href="https://thebiogrid.org" class="uri">https://thebiogrid.org</a>.
</p>
</div>
</div>
<div id="string" class="section level3 unnumbered">
<h3>STRING</h3>
<p>STRING combines experimental and computational evidence to infer PPIs.</p>
<div class="figure"><span id="fig:string-web"></span>
<img src="pic/string_web.png" alt="Screenshot of the STRING home page at https://string-db.org." width="512" />
<p class="caption">
Figure 4.10: Screenshot of the STRING home page at <a href="https://string-db.org" class="uri">https://string-db.org</a>.
</p>
</div>
</div>
</div>
<div id="other" class="section level2">
<h2><span class="header-section-number">4.7</span> Other</h2>
<div id="gene-ontology" class="section level3 unnumbered">
<h3>Gene Ontology</h3>
<p>One important difficulty when dealing with Biological structures (molecules, organelles, etc.) is the vocabulary we use to described them. As this type of information accumulated different researches would tend to use slighly different ways to describe the same thing. In other cases they can use similar words to describe different things. These nuances make the descriptions often found in publications ambiguous making it difficult to determine the exact meaning of some assertion. This difficulty became more prominent once the amount of information increased dramatically with the advent of omics technologies, and the information about the biological stuff was the subject of computational analyses. There was a need for an standardized nomenclature, and that was the origin of the <a href="http://www.geneontology.org">Gene Ontology</a> project. The purpose of the Gene Ontology project is stated in it web page:</p>
<blockquote>
<p>The Gene Ontology (GO) project is a major bioinformatics initiative to develop a computational representation of our evolving knowledge of how genes encode biological functions at the molecular, cellular and tissue system levels. Biological systems are so complex that we need to rely on computers to represent this knowledge. The project has developed formal ontologies that represent over 40,000 biological concepts, and are constantly being revised to reflect new discoveries. To date, these concepts have been used to “annotate” gene functions based on experiments reported in over 100,000 peer-reviewed scientific papers.</p>
</blockquote>
<p>In particular:</p>
<blockquote>
<p>The Gene Ontology project provides a controlled vocabulary of terms for describing gene product characteristics and gene product annotation data from GO Consortium members, as well as tools to access and process these data.</p>
</blockquote>
<div class="figure"><span id="fig:go-web"></span>
<img src="pic/go_web.png" alt="Screenshot of the Gene Ontology home page at http://www.geneontology.org." width="584" />
<p class="caption">
Figure 4.11: Screenshot of the Gene Ontology home page at <a href="http://www.geneontology.org" class="uri">http://www.geneontology.org</a>.
</p>
</div>
</div>
</div>
<div id="bioformatics-software" class="section level2">
<h2><span class="header-section-number">4.8</span> Bioformatics software</h2>
<div id="sequence-analysis-1" class="section level3">
<h3><span class="header-section-number">4.8.1</span> Sequence analysis</h3>
<div id="blast-1" class="section level4 unnumbered">
<h4>Blast</h4>
<p>The <a href="https://blast.ncbi.nlm.nih.gov">Blast</a> web page at NCBI is the main point of access to this versitile tool. It can be used to search for sequence similarity against any of the sequence databases at the NCBI.</p>
<div class="figure"><span id="fig:blast-web"></span>
<img src="pic/blast_web.png" alt="Screenshot of the Blast home page at https://blast.ncbi.nlm.nih.gov." width="606" />
<p class="caption">
Figure 4.12: Screenshot of the Blast home page at <a href="https://blast.ncbi.nlm.nih.gov" class="uri">https://blast.ncbi.nlm.nih.gov</a>.
</p>
</div>
</div>
<div id="hmmer" class="section level4 unnumbered">
<h4>HMMER</h4>
<p><a href="http://hmmer.org">HMMER</a> implements Hidden Markov Models for the analysis of biological sequences. It can learn models representing motifs, domains or entire sequences. These models are used to search for sequences matching the models in databases. It can also be used to annotate the domains/motifs present in a sequence using a database of domains. PFAM uses HMMER and manually curated MSA of domains to generate a database of protein families (PFAM). The use of HMM for sequence alignment is described in detail in the excellent book <a href="learning-bioinformatics.html#biological-sequence-analysis-probabilistic-models-of-proteins-and-nucleic-acids">Biological sequence analysis: Probabilistic models of proteins and nucleic acids</a>.</p>
<div class="figure"><span id="fig:hmmer-web"></span>
<img src="pic/hmmer_web.png" alt="Screenshot of the Hmmer home page at https://www.ebi.ac.uk/Tools/hmmer." width="598" />
<p class="caption">
Figure 4.13: Screenshot of the Hmmer home page at <a href="https://www.ebi.ac.uk/Tools/hmmer" class="uri">https://www.ebi.ac.uk/Tools/hmmer</a>.
</p>
</div>
</div>
<div id="jalview" class="section level4 unnumbered">
<h4>Jalview</h4>
<p><a href="http://www.jalview.org">Jalview</a> is a software for sequence analysis. It can visualize sequences, submit them to multiple sequence aligment servers, visualize the resulting MSA, create basic phylogenetic trees and visualize associated structures.</p>
<div class="figure"><span id="fig:jalview-web"></span>
<img src="pic/jalview_web.png" alt="Screenshot of the Jalview home page at http://www.jalview.org." width="608" />
<p class="caption">
Figure 4.14: Screenshot of the Jalview home page at <a href="http://www.jalview.org" class="uri">http://www.jalview.org</a>.
</p>
</div>
</div>
<div id="meme" class="section level4 unnumbered">
<h4>MEME</h4>
<p>The <a href="http://meme-suite.org">MEME-suite</a> is a collection of tools for the discovery of motifs in proteins, DNA and RNA sequences. If can find de-novo motifs (e.g. MEME) or search for known motifs from a motif database (e.g. MAST, FIMO, etc.)</p>
<div class="figure"><span id="fig:meme-web"></span>
<img src="pic/meme_web.png" alt="Screenshot of the MEME-suite home page at http://meme-suite.org." width="424" />
<p class="caption">
Figure 4.15: Screenshot of the MEME-suite home page at <a href="http://meme-suite.org" class="uri">http://meme-suite.org</a>.
</p>
</div>
</div>
</div>
<div id="next-generation-sequencing-1" class="section level3">
<h3><span class="header-section-number">4.8.2</span> Next generation sequencing</h3>
<div id="bowtie" class="section level4 unnumbered">
<h4>Bowtie</h4>
</div>
<div id="samtools" class="section level4 unnumbered">
<h4>Samtools</h4>
</div>
</div>
<div id="general-frameworks" class="section level3">
<h3><span class="header-section-number">4.8.3</span> General frameworks</h3>
<div id="bioconductor" class="section level4 unnumbered">
<h4>Bioconductor</h4>
<p><a href="http://www.bioconductor.org">Bioconductor</a> is a project providing R packages for the anlysis of omics datasets in the <a href="https://www.r-project.org">R software</a>. R runs on Windows, Macs and Linux computers.</p>
<blockquote>
<p>Bioconductor provides tools for the analysis and comprehension of high-throughput genomic data. Bioconductor uses the R statistical programming language, and is open source and open development.</p>
</blockquote>
<div class="figure"><span id="fig:bioconductor-web"></span>
<img src="pic/bioconductor_web.png" alt="Screenshot of the Bioconductor home page at http://www.bioconductor.org." width="476" />
<p class="caption">
Figure 4.16: Screenshot of the Bioconductor home page at <a href="http://www.bioconductor.org" class="uri">http://www.bioconductor.org</a>.
</p>
</div>
</div>
<div id="cytoscape" class="section level4 unnumbered">
<h4>Cytoscape</h4>
<p><a href="http://www.cytoscape.org">Cytoscape</a> is a software for the analysis and visualization of networks. It enables to import network information from online databases and integrate it with omics data like for example gene expression profiles. Cytoscape runs on Windows, Macs and Linux computers.</p>
<div class="figure"><span id="fig:cytoscape-hiv"></span>
<img src="pic/cytoscape_hiv.png" alt="Screenshot of Cytoscape showing the HIV sample dataset http://www.cytoscape.org." width="650" />
<p class="caption">
Figure 4.17: Screenshot of Cytoscape showing the HIV sample dataset <a href="http://www.cytoscape.org" class="uri">http://www.cytoscape.org</a>.
</p>
</div>
</div>
<div id="galaxy" class="section level4 unnumbered">
<h4>Galaxy</h4>
<p>The <a href="https://galaxyproject.org/tutorials/g101">Galaxy</a> project provides a web platform for the analysis of NGS data. In their own words:</p>
<blockquote>
<p>Galaxy software framework is an open-source application (distributed under the permissive Academic Free License. Its goal is to develop and maintain a system that enables researchers without informatics expertise to perform computational analyses through the web. A user interacts with Galaxy through the web by uploading and analyzing the data. Galaxy interacts with underlying computational infrastructure (servers that run the analyses and disks that store the data) without exposing it to the user.</p>
</blockquote>
<div class="figure"><span id="fig:gakaxy-web"></span>
<img src="pic/galaxy_web.png" alt="Screenshot of the main Galaxy site at http://usegalaxy.org." width="746" />
<p class="caption">
Figure 4.18: Screenshot of the main Galaxy site at <a href="http://usegalaxy.org" class="uri">http://usegalaxy.org</a>.
</p>
</div>
</div>
</div>
</div>
<div id="file-formats" class="section level2">
<h2><span class="header-section-number">4.9</span> File formats</h2>
<div id="fasta" class="section level3 unnumbered">
<h3>FASTA</h3>
<p>FASTA is a very simple file format used to store information about biological sequences, including protein, DNA and RNA. Each entry consists of the <code>&gt;</code> character followed by a sequence <em>Id</em> and a description, separated from the <em>Id</em> by a space character. The sequence itself goes in the next lines. Often FASTA files are formatted with a maximum number of residues per line (e.g. 60), which helps when visualizing the information in a regular text editor.</p>
<pre><code>&gt;seq1 This is seq1
AGTTAGGATTTGCGCCATT</code></pre>
</div>
<div id="fastq" class="section level3 unnumbered">
<h3>FASTQ</h3>
<p><a href="https://en.wikipedia.org/wiki/FASTQ_format">FASTQ</a> is a format used to store information from NGS technologies. In particular, stores the sequences of the reads in addition to information about the quality of sequencing. Each entry starts with the <code>@</code> character. That line also contains other information like the read identifier. The next line is the sequence information. Next there is a line with the <code>+</code> character, and then is the quality information encoded as a <a href="https://en.wikipedia.org/wiki/Phred_quality_score">phred score</a> of the same length as the sequence.</p>
<pre><code>@K00211:114:H52N5BBXX:1:2125:22252:26301
TTCCTATATAATCTTTATTTCTGGAAGTTAAAGTAGTTACAGCAATATACAAAAACAAACAACAACAAAAAACAACCCACAATAATATAAATTTTTACAC
+
AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJJJJJJJJJJJJJJJJJ</code></pre>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="omics-analysis-1.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="reproducible-research.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
