<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Introduction to Bioinformatics</title>
  <meta name="description" content="Material covered in the <em>Introduction to Bioinformatics</em> intensive course held in Nagoya University, Dec 13-14, 2017.">
  <meta name="generator" content="bookdown 0.5 and GitBook 2.6.7">

  <meta property="og:title" content="Introduction to Bioinformatics" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Material covered in the <em>Introduction to Bioinformatics</em> intensive course held in Nagoya University, Dec 13-14, 2017." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Introduction to Bioinformatics" />
  
  <meta name="twitter:description" content="Material covered in the <em>Introduction to Bioinformatics</em> intensive course held in Nagoya University, Dec 13-14, 2017." />
  

<meta name="author" content="Diego Diez">


<meta name="date" content="2017-12-10">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="prerequisites.html">
<link rel="next" href="sequence-analysis.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Introduction to Bioinformatics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About</a></li>
<li class="chapter" data-level="" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i>Prerequisites</a></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="intro.html"><a href="intro.html#intro-sequence-analysis"><i class="fa fa-check"></i><b>1.1</b> Sequence analysis</a></li>
<li class="chapter" data-level="1.2" data-path="intro.html"><a href="intro.html#omics-analysis"><i class="fa fa-check"></i><b>1.2</b> Omics analysis</a></li>
<li class="chapter" data-level="1.3" data-path="intro.html"><a href="intro.html#structural-analysis"><i class="fa fa-check"></i><b>1.3</b> Structural analysis</a></li>
<li class="chapter" data-level="1.4" data-path="intro.html"><a href="intro.html#phylogenetic-analysis"><i class="fa fa-check"></i><b>1.4</b> Phylogenetic analysis</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="sequence-analysis.html"><a href="sequence-analysis.html"><i class="fa fa-check"></i><b>2</b> Sequence analysis</a><ul>
<li class="chapter" data-level="2.1" data-path="sequence-analysis.html"><a href="sequence-analysis.html#pairwise-alignment"><i class="fa fa-check"></i><b>2.1</b> Pairwise alignment</a></li>
<li class="chapter" data-level="2.2" data-path="sequence-analysis.html"><a href="sequence-analysis.html#scoring-matrices"><i class="fa fa-check"></i><b>2.2</b> Scoring matrices</a></li>
<li class="chapter" data-level="2.3" data-path="sequence-analysis.html"><a href="sequence-analysis.html#blast"><i class="fa fa-check"></i><b>2.3</b> Blast</a></li>
<li class="chapter" data-level="2.4" data-path="sequence-analysis.html"><a href="sequence-analysis.html#multiple-sequence-alignment"><i class="fa fa-check"></i><b>2.4</b> Multiple sequence alignment</a></li>
<li class="chapter" data-level="2.5" data-path="sequence-analysis.html"><a href="sequence-analysis.html#tree-methods"><i class="fa fa-check"></i><b>2.5</b> Tree methods</a></li>
<li class="chapter" data-level="2.6" data-path="sequence-analysis.html"><a href="sequence-analysis.html#domains-and-motifs"><i class="fa fa-check"></i><b>2.6</b> Domains and motifs</a></li>
<li class="chapter" data-level="2.7" data-path="sequence-analysis.html"><a href="sequence-analysis.html#phylogenetics"><i class="fa fa-check"></i><b>2.7</b> Phylogenetics</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html"><i class="fa fa-check"></i><b>3</b> Omics analysis</a><ul>
<li class="chapter" data-level="3.1" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#microarrays"><i class="fa fa-check"></i><b>3.2</b> Microarrays</a><ul>
<li class="chapter" data-level="3.2.1" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#expression-microarrays"><i class="fa fa-check"></i><b>3.2.1</b> Expression microarrays</a></li>
<li class="chapter" data-level="3.2.2" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#analysis-of-microarray-data"><i class="fa fa-check"></i><b>3.2.2</b> Analysis of microarray data</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#next-generation-sequencing"><i class="fa fa-check"></i><b>3.3</b> Next-generation sequencing</a><ul>
<li class="chapter" data-level="3.3.1" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#rna-seq"><i class="fa fa-check"></i><b>3.3.1</b> RNA-seq</a></li>
<li class="chapter" data-level="3.3.2" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#example-stat3-expression-in-macrophages-exposed-to-il10"><i class="fa fa-check"></i><b>3.3.2</b> Example: Stat3 expression in macrophages exposed to IL10</a></li>
<li class="chapter" data-level="3.3.3" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#chip-seq"><i class="fa fa-check"></i><b>3.3.3</b> ChIP-seq</a></li>
<li class="chapter" data-level="3.3.4" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#example-stat3-binding-in-macrophages-exposed-to-il10"><i class="fa fa-check"></i><b>3.3.4</b> Example: Stat3 binding in macrophages exposed to IL10</a></li>
<li class="chapter" data-level="3.3.5" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#other-ngs-technologies"><i class="fa fa-check"></i><b>3.3.5</b> Other NGS technologies</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#mass-spectromics-methods"><i class="fa fa-check"></i><b>3.4</b> Mass spectromics methods</a><ul>
<li class="chapter" data-level="3.4.1" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#ms"><i class="fa fa-check"></i><b>3.4.1</b> MS</a></li>
<li class="chapter" data-level="3.4.2" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#ap-ms"><i class="fa fa-check"></i><b>3.4.2</b> AP-MS</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#single-cell-omics"><i class="fa fa-check"></i><b>3.5</b> Single cell omics</a></li>
<li class="chapter" data-level="3.6" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#batch-effects"><i class="fa fa-check"></i><b>3.6</b> Batch effects</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html"><i class="fa fa-check"></i><b>4</b> Bioinformatics resources</a><ul>
<li class="chapter" data-level="4.1" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#gene-ontology"><i class="fa fa-check"></i><b>4.1</b> Gene Ontology</a></li>
<li class="chapter" data-level="4.2" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#sequence-databases"><i class="fa fa-check"></i><b>4.2</b> Sequence databases</a><ul>
<li class="chapter" data-level="4.2.1" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#ncbi"><i class="fa fa-check"></i><b>4.2.1</b> NCBI</a></li>
<li class="chapter" data-level="4.2.2" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#ensembl"><i class="fa fa-check"></i><b>4.2.2</b> EnsEMBL</a></li>
<li class="chapter" data-level="4.2.3" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#uniprot"><i class="fa fa-check"></i><b>4.2.3</b> Uniprot</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#gene-expression-databases"><i class="fa fa-check"></i><b>4.3</b> Gene expression databases</a><ul>
<li class="chapter" data-level="4.3.1" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#geo-gene-expression-omnibus"><i class="fa fa-check"></i><b>4.3.1</b> GEO: Gene Expression Omnibus</a></li>
<li class="chapter" data-level="4.3.2" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#arrayexpress"><i class="fa fa-check"></i><b>4.3.2</b> ArrayExpress</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#pathway-databases"><i class="fa fa-check"></i><b>4.4</b> Pathway databases</a><ul>
<li class="chapter" data-level="4.4.1" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#kegg"><i class="fa fa-check"></i><b>4.4.1</b> KEGG</a></li>
<li class="chapter" data-level="4.4.2" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#reactome"><i class="fa fa-check"></i><b>4.4.2</b> Reactome</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#domain-databases"><i class="fa fa-check"></i><b>4.5</b> Domain databases</a><ul>
<li class="chapter" data-level="4.5.1" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#pfam"><i class="fa fa-check"></i><b>4.5.1</b> Pfam</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#motif-databases"><i class="fa fa-check"></i><b>4.6</b> Motif databases</a><ul>
<li class="chapter" data-level="4.6.1" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#jaspar"><i class="fa fa-check"></i><b>4.6.1</b> Jaspar</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#interaction-networks"><i class="fa fa-check"></i><b>4.7</b> Interaction networks</a><ul>
<li class="chapter" data-level="4.7.1" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#biogrid"><i class="fa fa-check"></i><b>4.7.1</b> BioGRID</a></li>
<li class="chapter" data-level="4.7.2" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#string"><i class="fa fa-check"></i><b>4.7.2</b> STRING</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#bioformatics-software"><i class="fa fa-check"></i><b>4.8</b> Bioformatics software</a><ul>
<li class="chapter" data-level="4.8.1" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#blast-1"><i class="fa fa-check"></i><b>4.8.1</b> Blast</a></li>
<li class="chapter" data-level="4.8.2" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#jalview"><i class="fa fa-check"></i><b>4.8.2</b> Jalview</a></li>
<li class="chapter" data-level="4.8.3" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#meme"><i class="fa fa-check"></i><b>4.8.3</b> MEME</a></li>
<li class="chapter" data-level="4.8.4" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#bioconductor"><i class="fa fa-check"></i><b>4.8.4</b> Bioconductor</a></li>
<li class="chapter" data-level="4.8.5" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#cytoscape"><i class="fa fa-check"></i><b>4.8.5</b> Cytoscape</a></li>
</ul></li>
<li class="chapter" data-level="4.9" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#file-formats"><i class="fa fa-check"></i><b>4.9</b> File formats</a><ul>
<li class="chapter" data-level="4.9.1" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#fasta"><i class="fa fa-check"></i><b>4.9.1</b> FASTA</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="reproducible-research.html"><a href="reproducible-research.html"><i class="fa fa-check"></i><b>5</b> Reproducible research</a><ul>
<li class="chapter" data-level="5.1" data-path="reproducible-research.html"><a href="reproducible-research.html#git"><i class="fa fa-check"></i><b>5.1</b> Git</a><ul>
<li class="chapter" data-level="5.1.1" data-path="reproducible-research.html"><a href="reproducible-research.html#github"><i class="fa fa-check"></i><b>5.1.1</b> Github</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="reproducible-research.html"><a href="reproducible-research.html#literate-programming"><i class="fa fa-check"></i><b>5.2</b> Literate programming</a><ul>
<li class="chapter" data-level="5.2.1" data-path="reproducible-research.html"><a href="reproducible-research.html#rmarkdown-notebooks"><i class="fa fa-check"></i><b>5.2.1</b> RMarkdown notebooks</a></li>
<li class="chapter" data-level="5.2.2" data-path="reproducible-research.html"><a href="reproducible-research.html#jupyter-notebooks"><i class="fa fa-check"></i><b>5.2.2</b> Jupyter notebooks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html"><i class="fa fa-check"></i><b>6</b> Learning Bioinformatics</a><ul>
<li class="chapter" data-level="6.1" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#books"><i class="fa fa-check"></i><b>6.1</b> Books</a><ul>
<li class="chapter" data-level="6.1.1" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#sequence-analysis-1"><i class="fa fa-check"></i><b>6.1.1</b> Sequence analysis</a></li>
<li class="chapter" data-level="6.1.2" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#omics-technologies"><i class="fa fa-check"></i><b>6.1.2</b> Omics technologies</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#online-resources"><i class="fa fa-check"></i><b>6.2</b> Online resources</a><ul>
<li class="chapter" data-level="6.2.1" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#learning-how-to-use-the-command-line"><i class="fa fa-check"></i><b>6.2.1</b> Learning how to use the command line</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#community-resources"><i class="fa fa-check"></i><b>6.3</b> Community resources</a><ul>
<li class="chapter" data-level="6.3.1" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#biostars"><i class="fa fa-check"></i><b>6.3.1</b> Biostars</a></li>
<li class="chapter" data-level="6.3.2" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#bioinformatics-channel-at-stackexchange"><i class="fa fa-check"></i><b>6.3.2</b> Bioinformatics channel at Stackexchange</a></li>
<li class="chapter" data-level="6.3.3" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#bioconductor-support-site"><i class="fa fa-check"></i><b>6.3.3</b> Bioconductor support site</a></li>
<li class="chapter" data-level="6.3.4" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#how-to-use-the-support-sites"><i class="fa fa-check"></i><b>6.3.4</b> How to use the support sites</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to Bioinformatics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="intro" class="section level1">
<h1><span class="header-section-number">Chapter 1</span> Introduction</h1>
<p>Bioinformatics can be defined as the use of computational methods to answer questions in Biology. Another way to define it is as the field concerned with the development of such tools. Another related name is Computational Biologist, which refers to a Biologist that use mainly Bioinformatics tools to tackle questions in Biology. All these terms can be confusing but here we will put our focus on the Biologist point of view. From that perspective Bioinformatics is just a tool, in the same way as Molecular Biology is, that can help us to get insight into Biology.</p>
<p>Another confusion comes from the overlap with other fields like Statistics and Computer Science. This is because researchers that use Bioinformatics tools may inevitably end up using also methods and tools from Statistics, Computer Science, Machine learning, etc. Here we will concentrate on the description of mainly purely Bioinformatics tools. However, it is unavoidable that reference and/or knowledge of these other fields will be required to have some understanding of how Bioinformatics tools work, and also to be able to interpret the results from Bioinformatics tools.</p>
<p>Bioinformatics is a very broad terms covering and overlapping with other previously existing fields. Some of the topics touched by Bioinformatics include:</p>
<ul>
<li>Sequence analysis</li>
<li>Phylogenetic analysis</li>
<li>Omics analysis</li>
<li>Structural analysis</li>
</ul>
<p>Below we will briefly describe each of these topics. In this introductory course, however, we will focus on two topics, sequence analysis and omics analysis.</p>
<div id="intro-sequence-analysis" class="section level2">
<h2><span class="header-section-number">1.1</span> Sequence analysis</h2>
<p>The main goal of sequence analysis is two answer questions regarding DNA, RNA and/or protein sequences. For example, we could be presented with an unknown sequence, like the one shown here:</p>
<pre><code>MTEYKLVVVGAGGVGKSALTIQLIQNHFVDEYDPTIEDSYRKQVVIDGETCLLDILDTAG
QEEYSAMRDQYMRTGEGFLCVFAINNTKSFEDIHHYREQIKRVKDSEDVPMVLVGNKCDL
PSRTVDTKQAQDLARSYGIPFIETSAKTRQRVEDAFYTLVREIRQYRLKKISKEEKTPGC
VKIKKCIIM</code></pre>
<p>The question is what can we learn about this unknown sequence identity from the sequence information alone. In particular, we can ask:</p>
<ul>
<li>Are there any other similar sequences?</li>
<li>What parts of the sequences are similar and which one are not?</li>
<li>Can we see a conservation pattern among a set of sequences?</li>
<li>What is its function?</li>
</ul>
<p>Sequence analysis deals with the methods used to answer some of these questions.</p>
</div>
<div id="omics-analysis" class="section level2">
<h2><span class="header-section-number">1.2</span> Omics analysis</h2>
<p>In omics analysis we are concerned with the manipulation, processing and analysis of high-throughtput data from omics technologies, including microarrays, next generation sequencing (NGS) and mass-spectromemtry (MS). High-throughput here means that we are concerned with measuring the expression of, for example, all the transcripts in the cell, not just a small number of them. We may also be interested on the information about all the proteins, or all the metabolites, etc. Omics technologies enables this to be done, and generate large datasets with information about all the molecular components of the cell.</p>
<p>For example, we can consider an study in which the effect of some drug on the expression profile of some cells will be evaluated. Mice are treated with either the drug or some placebo. Cells are isolated from the tissue and RNA is extracted. After amplification and convertion into cDNA we hybridize the samples to microarrays containing probes for &gt;55 thousand transcripts. This will give us an estimate of the expression profile for 55 thousand transcripts. If we have initially 3 samples per group, we will obtain a 55,000 x 6 matrix of expression values.</p>
<p>Omics analysis deals with problems associated with the manipulation, processing and analysis of omics datasets. How can we process and analyze this data in a way that avoids biases and gives us some information about the biological question of interest?</p>
</div>
<div id="structural-analysis" class="section level2">
<h2><span class="header-section-number">1.3</span> Structural analysis</h2>
<p>Structural bioinformatics deals with methods predicting the structure of biopolymers and their interactions with one another as well as with small molecules. It can be very important in order to decipher the mechanism of action of some drugs.</p>
</div>
<div id="phylogenetic-analysis" class="section level2">
<h2><span class="header-section-number">1.4</span> Phylogenetic analysis</h2>
<p>Phylogenetic analysis is concerned with the evolutionary relationships between biological sequences and what can that tell us about the evolution of species. It uses information directly related with the sequence analysis methods described above, and includes additional methods to talckle specific questions in the field.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="prerequisites.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="sequence-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
