<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Introduction to Bioinformatics</title>
  <meta name="description" content="Material covered in the <em>Introduction to Bioinformatics</em> intensive course held in Nagoya University, Dec 13-14, 2017.">
  <meta name="generator" content="bookdown 0.5 and GitBook 2.6.7">

  <meta property="og:title" content="Introduction to Bioinformatics" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Material covered in the <em>Introduction to Bioinformatics</em> intensive course held in Nagoya University, Dec 13-14, 2017." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Introduction to Bioinformatics" />
  
  <meta name="twitter:description" content="Material covered in the <em>Introduction to Bioinformatics</em> intensive course held in Nagoya University, Dec 13-14, 2017." />
  

<meta name="author" content="Diego Diez">


<meta name="date" content="2017-12-12">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="reproducible-research.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Introduction to Bioinformatics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About</a></li>
<li class="chapter" data-level="" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i>Prerequisites</a></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="intro.html"><a href="intro.html#intro-sequence-analysis"><i class="fa fa-check"></i><b>1.1</b> Sequence analysis</a></li>
<li class="chapter" data-level="1.2" data-path="intro.html"><a href="intro.html#omics-analysis"><i class="fa fa-check"></i><b>1.2</b> Omics analysis</a></li>
<li class="chapter" data-level="1.3" data-path="intro.html"><a href="intro.html#structural-analysis"><i class="fa fa-check"></i><b>1.3</b> Structural analysis</a></li>
<li class="chapter" data-level="1.4" data-path="intro.html"><a href="intro.html#phylogenetic-analysis"><i class="fa fa-check"></i><b>1.4</b> Phylogenetic analysis</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="sequence-analysis.html"><a href="sequence-analysis.html"><i class="fa fa-check"></i><b>2</b> Sequence analysis</a><ul>
<li class="chapter" data-level="2.1" data-path="sequence-analysis.html"><a href="sequence-analysis.html#pairwise-alignment"><i class="fa fa-check"></i><b>2.1</b> Pairwise alignment</a></li>
<li class="chapter" data-level="2.2" data-path="sequence-analysis.html"><a href="sequence-analysis.html#scoring-matrices"><i class="fa fa-check"></i><b>2.2</b> Scoring matrices</a></li>
<li class="chapter" data-level="2.3" data-path="sequence-analysis.html"><a href="sequence-analysis.html#blast"><i class="fa fa-check"></i><b>2.3</b> Blast</a></li>
<li class="chapter" data-level="2.4" data-path="sequence-analysis.html"><a href="sequence-analysis.html#multiple-sequence-alignment"><i class="fa fa-check"></i><b>2.4</b> Multiple sequence alignment</a></li>
<li class="chapter" data-level="2.5" data-path="sequence-analysis.html"><a href="sequence-analysis.html#tree-methods"><i class="fa fa-check"></i><b>2.5</b> Tree methods</a></li>
<li class="chapter" data-level="2.6" data-path="sequence-analysis.html"><a href="sequence-analysis.html#domains-and-motifs"><i class="fa fa-check"></i><b>2.6</b> Domains and motifs</a></li>
<li class="chapter" data-level="2.7" data-path="sequence-analysis.html"><a href="sequence-analysis.html#phylogenetics"><i class="fa fa-check"></i><b>2.7</b> Phylogenetics</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html"><i class="fa fa-check"></i><b>3</b> Omics analysis</a><ul>
<li class="chapter" data-level="3.1" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#introduction"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#microarrays"><i class="fa fa-check"></i><b>3.2</b> Microarrays</a></li>
<li class="chapter" data-level="3.3" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#next-generation-sequencing"><i class="fa fa-check"></i><b>3.3</b> Next-generation sequencing</a><ul>
<li class="chapter" data-level="3.3.1" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#rna-seq"><i class="fa fa-check"></i><b>3.3.1</b> RNA-seq</a></li>
<li class="chapter" data-level="3.3.2" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#chip-seq"><i class="fa fa-check"></i><b>3.3.2</b> ChIP-seq</a></li>
<li class="chapter" data-level="3.3.3" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#other-ngs-technologies"><i class="fa fa-check"></i><b>3.3.3</b> Other NGS technologies</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#mass-spectromics-methods"><i class="fa fa-check"></i><b>3.4</b> Mass spectromics methods</a><ul>
<li class="chapter" data-level="3.4.1" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#ms"><i class="fa fa-check"></i><b>3.4.1</b> MS</a></li>
<li class="chapter" data-level="3.4.2" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#ap-ms"><i class="fa fa-check"></i><b>3.4.2</b> AP-MS</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="omics-analysis-1.html"><a href="omics-analysis-1.html#single-cell-omics"><i class="fa fa-check"></i><b>3.5</b> Single cell omics</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html"><i class="fa fa-check"></i><b>4</b> Bioinformatics resources</a><ul>
<li class="chapter" data-level="4.1" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#sequence-databases"><i class="fa fa-check"></i><b>4.1</b> Sequence databases</a><ul>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#ncbi"><i class="fa fa-check"></i>NCBI</a></li>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#ensembl"><i class="fa fa-check"></i>EnsEMBL</a></li>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#uniprot"><i class="fa fa-check"></i>Uniprot</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#gene-expression-databases"><i class="fa fa-check"></i><b>4.2</b> Gene expression databases</a><ul>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#geo-gene-expression-omnibus"><i class="fa fa-check"></i>GEO: Gene Expression Omnibus</a></li>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#arrayexpress"><i class="fa fa-check"></i>ArrayExpress</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#pathway-databases"><i class="fa fa-check"></i><b>4.3</b> Pathway databases</a><ul>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#kegg"><i class="fa fa-check"></i>KEGG</a></li>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#reactome"><i class="fa fa-check"></i>Reactome</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#domain-databases"><i class="fa fa-check"></i><b>4.4</b> Domain databases</a><ul>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#pfam"><i class="fa fa-check"></i>Pfam</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#motif-databases"><i class="fa fa-check"></i><b>4.5</b> Motif databases</a><ul>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#jaspar"><i class="fa fa-check"></i>Jaspar</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#interaction-databases"><i class="fa fa-check"></i><b>4.6</b> Interaction databases</a><ul>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#biogrid"><i class="fa fa-check"></i>BioGRID</a></li>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#string"><i class="fa fa-check"></i>STRING</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#other"><i class="fa fa-check"></i><b>4.7</b> Other</a><ul>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#gene-ontology"><i class="fa fa-check"></i>Gene Ontology</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#bioformatics-software"><i class="fa fa-check"></i><b>4.8</b> Bioformatics software</a><ul>
<li class="chapter" data-level="4.8.1" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#sequence-analysis-1"><i class="fa fa-check"></i><b>4.8.1</b> Sequence analysis</a></li>
<li class="chapter" data-level="4.8.2" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#next-generation-sequencing-1"><i class="fa fa-check"></i><b>4.8.2</b> Next generation sequencing</a></li>
<li class="chapter" data-level="4.8.3" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#general-frameworks"><i class="fa fa-check"></i><b>4.8.3</b> General frameworks</a></li>
</ul></li>
<li class="chapter" data-level="4.9" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#file-formats"><i class="fa fa-check"></i><b>4.9</b> File formats</a><ul>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#fasta"><i class="fa fa-check"></i>FASTA</a></li>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#fastq"><i class="fa fa-check"></i>Fastq</a></li>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#sam"><i class="fa fa-check"></i>SAM</a></li>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#bed"><i class="fa fa-check"></i>BED</a></li>
<li class="chapter" data-level="" data-path="bioinformatics-resources.html"><a href="bioinformatics-resources.html#gff"><i class="fa fa-check"></i>GFF</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="reproducible-research.html"><a href="reproducible-research.html"><i class="fa fa-check"></i><b>5</b> Reproducible research</a><ul>
<li class="chapter" data-level="5.1" data-path="reproducible-research.html"><a href="reproducible-research.html#git"><i class="fa fa-check"></i><b>5.1</b> Git</a><ul>
<li class="chapter" data-level="" data-path="reproducible-research.html"><a href="reproducible-research.html#github"><i class="fa fa-check"></i>Github</a></li>
<li class="chapter" data-level="" data-path="reproducible-research.html"><a href="reproducible-research.html#gitlab"><i class="fa fa-check"></i>Gitlab</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="reproducible-research.html"><a href="reproducible-research.html#literate-programming"><i class="fa fa-check"></i><b>5.2</b> Literate programming</a><ul>
<li class="chapter" data-level="" data-path="reproducible-research.html"><a href="reproducible-research.html#rmarkdown-notebooks"><i class="fa fa-check"></i>RMarkdown notebooks</a></li>
<li class="chapter" data-level="" data-path="reproducible-research.html"><a href="reproducible-research.html#jupyter-notebooks"><i class="fa fa-check"></i>Jupyter notebooks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html"><i class="fa fa-check"></i><b>6</b> Learning Bioinformatics</a><ul>
<li class="chapter" data-level="6.1" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#books"><i class="fa fa-check"></i><b>6.1</b> Books</a><ul>
<li class="chapter" data-level="6.1.1" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#sequence-analysis-2"><i class="fa fa-check"></i><b>6.1.1</b> Sequence analysis</a></li>
<li class="chapter" data-level="6.1.2" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#omics-technologies"><i class="fa fa-check"></i><b>6.1.2</b> Omics technologies</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#online-resources"><i class="fa fa-check"></i><b>6.2</b> Online resources</a><ul>
<li class="chapter" data-level="6.2.1" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#online-courses"><i class="fa fa-check"></i><b>6.2.1</b> Online courses</a></li>
<li class="chapter" data-level="6.2.2" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#youtube"><i class="fa fa-check"></i><b>6.2.2</b> YouTube</a></li>
<li class="chapter" data-level="6.2.3" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#how-to-use-the-command-line"><i class="fa fa-check"></i><b>6.2.3</b> How to use the command line</a></li>
<li class="chapter" data-level="6.2.4" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#programming-languages"><i class="fa fa-check"></i><b>6.2.4</b> Programming languages</a></li>
<li class="chapter" data-level="6.2.5" data-path="learning-bioinformatics.html"><a href="learning-bioinformatics.html#support-communities"><i class="fa fa-check"></i><b>6.2.5</b> Support Communities</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to Bioinformatics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="learning-bioinformatics" class="section level1">
<h1><span class="header-section-number">Chapter 6</span> Learning Bioinformatics</h1>
<div id="books" class="section level2">
<h2><span class="header-section-number">6.1</span> Books</h2>
<div id="sequence-analysis-2" class="section level3">
<h3><span class="header-section-number">6.1.1</span> Sequence analysis</h3>
<div id="blast-an-essential-guide-to-the-basic-local-alignment-search-tool" class="section level4 unnumbered">
<h4>BLAST: An essential guide to the Basic Local Alignment Search Tool</h4>
<p>Introductory book to the topic of sequence aligments in general and Blast in particular. Includes Perl code.</p>
<div class="figure"><span id="fig:unnamed-chunk-12"></span>
<img src="pic/blast_book.jpg" alt="Blast. Ian Korf, ... O'Reilly, 2003" width="250" />
<p class="caption">
Figure 6.1: Blast. Ian Korf, … O’Reilly, 2003
</p>
</div>
</div>
<div id="biological-sequence-analysis-probabilistic-models-of-proteins-and-nucleic-acids" class="section level4 unnumbered">
<h4>Biological sequence analysis: Probabilistic models of proteins and nucleic acids</h4>
<p>Advance level understanding of the theory of sequence aligment and Hidden Markov Models (HMM).</p>
<div class="figure"><span id="fig:unnamed-chunk-13"></span>
<img src="pic/hmm_book.jpg" alt="Biological sequence analysis: Probabilistic models of proteins and nucleic acids. R. Durbin, S. Eddy, ... Cambridge, 1998" width="174" />
<p class="caption">
Figure 6.2: Biological sequence analysis: Probabilistic models of proteins and nucleic acids. R. Durbin, S. Eddy, … Cambridge, 1998
</p>
</div>
</div>
<div id="sequence-analysis-in-a-nutshell-a-guide-to-tools-and-databases" class="section level4">
<h4><span class="header-section-number">6.1.1.1</span> Sequence Analysis in a Nutshell: A Guide to Tools and Databases</h4>
<p>General introduction to many of the sequence analysis tools covered in this course.</p>
<div class="figure"><span id="fig:unnamed-chunk-14"></span>
<img src="pic/seq_analysis_book.png" alt="Sequence Analysis in a Nutshell: A Guide to Tools and Databases. S. Markel, D. Leon , ... Oreilly, 2003" width="298" />
<p class="caption">
Figure 6.3: Sequence Analysis in a Nutshell: A Guide to Tools and Databases. S. Markel, D. Leon , … Oreilly, 2003
</p>
</div>
</div>
<div id="molecular-evolution-and-phylogenetics" class="section level4 unnumbered">
<h4>Molecular evolution and phylogenetics</h4>
<p>Introductory level book on the topic of inferring phylogenies from sequence information.</p>
<div class="figure"><span id="fig:unnamed-chunk-15"></span>
<img src="pic/phylogenetics_book.jpg" alt="Molecular evolution and phylogenetics. M. Nei, S. Kumar. Oxford University Press, 2000." width="171" />
<p class="caption">
Figure 6.4: Molecular evolution and phylogenetics. M. Nei, S. Kumar. Oxford University Press, 2000.
</p>
</div>
</div>
</div>
<div id="omics-technologies" class="section level3">
<h3><span class="header-section-number">6.1.2</span> Omics technologies</h3>
</div>
</div>
<div id="online-resources" class="section level2">
<h2><span class="header-section-number">6.2</span> Online resources</h2>
<div id="online-courses" class="section level3">
<h3><span class="header-section-number">6.2.1</span> Online courses</h3>
<p>There are many free online courses by instructors from reputable universities at web sites like <a href="https://www.edx.org">edX</a>. You can find many topics related to Bioinformatics, Computational Biology, Statistics and Data analysis.</p>
<div class="figure"><span id="fig:edx-web"></span>
<img src="pic/edx_web.png" alt="Screenshot of the edX home page at https://www.edx.org." width="634" />
<p class="caption">
Figure 6.5: Screenshot of the edX home page at <a href="https://www.edx.org" class="uri">https://www.edx.org</a>.
</p>
</div>
</div>
<div id="youtube" class="section level3">
<h3><span class="header-section-number">6.2.2</span> YouTube</h3>
<p>Many Bioinformatics resources are putting a lot of resources into teaching how to use their services, with an emphasis on newcomers. Some of them provide introductory tutorials and case studies in videos hosted in their YouTube channels. For example there are channels for <a href="https://www.youtube.com/user/NCBINLM">NCBI</a>, <a href="https://www.youtube.com/user/uniprotvideos">Uniprot</a>, <a href="https://www.youtube.com/user/EnsemblHelpdesk">EnsEMBL</a>, and <a href="https://www.youtube.com/user/bioconductor">Bioconductor</a>, to mention a few.</p>
<div class="figure"><span id="fig:youtube-uniprot"></span>
<img src="pic/youtube_uniprot.png" alt="Screenshot of the Uniprot YouTube channel home page at https://www.youtube.com/user/uniprotvideos." width="734" />
<p class="caption">
Figure 6.6: Screenshot of the Uniprot YouTube channel home page at <a href="https://www.youtube.com/user/uniprotvideos" class="uri">https://www.youtube.com/user/uniprotvideos</a>.
</p>
</div>
</div>
<div id="how-to-use-the-command-line" class="section level3">
<h3><span class="header-section-number">6.2.3</span> How to use the command line</h3>
<p>Often using Bioinformatics tools requires knowledge of how to use the command line, specially in Linux/Unix environments. <a href="http://linuxcommand.org/index.php">http://linuxcommand.org</a> has a nice introduction to the Linux command line. Another interesting resource is <a href="https://www.codecademy.com" class="uri">https://www.codecademy.com</a>, providing an interactive learning experience (It offers also paid courses). Also, <a href="https://ryanstutorials.net/bash-scripting-tutorial/">here</a> there is a tutorial on <em>Bash</em> scripting.</p>
</div>
<div id="programming-languages" class="section level3">
<h3><span class="header-section-number">6.2.4</span> Programming languages</h3>
<p>Learning a programming language is not absolutely necessary when using most of the bioinformatics tools described in this book. However, when analyzing high-throughput datasets it is neccessary to use them. In particular, <a href="https://www.python.org">Python</a> and <a href="https://www.r-project.org">R</a> are the most used languages in data science, including Bioinformatics. There are many free online resources to learn the basics of these programming languages. Check their web pages for tutorials for beginners.</p>
<div class="figure"><span id="fig:python-web"></span>
<img src="pic/python_web.png" alt="Screenshot of the Python home page at https://www.python.org." width="594" />
<p class="caption">
Figure 6.7: Screenshot of the Python home page at <a href="https://www.python.org" class="uri">https://www.python.org</a>.
</p>
</div>
<div class="figure"><span id="fig:r-web"></span>
<img src="pic/r_web.png" alt="Screenshot of the R home page at https://www.r-project.org." width="432" />
<p class="caption">
Figure 6.8: Screenshot of the R home page at <a href="https://www.r-project.org" class="uri">https://www.r-project.org</a>.
</p>
</div>
</div>
<div id="support-communities" class="section level3">
<h3><span class="header-section-number">6.2.5</span> Support Communities</h3>
<p>Learning Bioinformatics, as with any other complex topic, can be daunting. Often Bioinformatics software come with good and detailed documentation and. It is also possible to frequently find short tutorials that can help newcomers get started. It is important to get a <em>good</em> understanding of the methods being used when the goal is to publish your results. The Bioinformatics community can sometimes help in the learning process by providing solutions to the most frequently encountered problems. Here I will highlight some of the most relevant online communities for the Bioinformatician.</p>
<div id="biostars" class="section level4 unnumbered">
<h4>Biostars</h4>
<p><a href="http://biostars.org">Biostars</a> is an online community deboted to answering questions about <em>“bioinformatics, computational genomics and biological data analysis”</em>. Users can post questions, get answers in the same post. There is a rating system that enables to give credit to good questions and answers. It is possible to include comments for further clarification and discussion.</p>
</div>
<div id="bioinformatics-stack-exchange" class="section level4 unnumbered">
<h4>Bioinformatics Stack Exchange</h4>
<p>Another useful resource for solving general questions related to Bioinformatics can be the <a href="https://bioinformatics.stackexchange.com">Bioinformatics Stack Exchange</a> network. These site follows the same phylosohpy of Biostars (indeed, Biostars originated in the old version of Stack Exchange) but offers, in my opinion, a better user inteface. The site is still in beta but most of the Biostars community can be found also in this new site.</p>
</div>
<div id="bioconductor-support-site" class="section level4 unnumbered">
<h4>Bioconductor support site</h4>
<p>The <a href="https://support.bioconductor.org">Bioconductor support site</a> provides support for questions related to the use of Bioconductor, using a similar interface to Biostars.</p>
</div>
<div id="how-to-use-the-support-sites" class="section level4 unnumbered">
<h4>How to use the support sites</h4>
<p>It is important to remember that online support sites are usually driven by voluntarees that lend their time and expertise to help others. Because the number of questions put in one of these support sites can be very large, it is very helpful if the questions themeselves are written in a way that simplify the work of those willing to help us with our doubts. For example, it is important to write simple yet meaningful titles that summarize what our problem is. This way the community can easily know if the question is related to a topic about which they have expertise. When describing your issues, be as explicit as possible. And whenever possible, include a <a href="https://en.wikipedia.org/wiki/Minimal_Working_Example">minimal reproducible example</a>. This is particularly important if you have problems with programming code but it can also apply to using a web server. For example, if you had some problem using a web server to generate a sequence aligment, include all the steps you performed to obtain the error (if any) you obtained. If possible, when needed, include a minimal example dataset to reproduce the error. In this example, it could be a small file with a few sequences that trigger the error.</p>
<p>The particulars will depend of the software and problem you are having. But as a rule of thumb it is always helpful to put yourself in the place of those reading your question, and imagining whether that is enough for them to understand your problem.</p>

</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="reproducible-research.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(script.src))
      script.src  = script.src.replace(/^https?:/, '');
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
